{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 处理物种信息"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# 读取物种分类文件\n",
    "gtdb_file = '/storage/jufengLab/luogaoyang/metagenome_project/DSR/1_clean/bio_all_reads/GTDB_OUT/classify/gtdbtk.bac120.summary.tsv'  # 替换为实际文件路径\n",
    "gtdb_file = '/storage/jufengLab/luogaoyang/metagenome_project/DSR/1_clean/nonbio_all_reads/GTDB_OUT/classify/gtdbtk.bac120.summary.tsv'  # 替换为实际文件路径\n",
    "# gtdb_file = '/storage/jufengLab/luogaoyang/metagenome_project/DSR/1_clean/water_all_reads/GTDB_OUT/classify/gtdbtk.bac120.summary.tsv'  # 替换为实际文件路径\n",
    "# gtdb_file = '/storage/jufengLab/luogaoyang/metagenome_project/DSR/1_clean/wood_all_reads/GTDB_OUT/classify/gtdbtk.bac120.summary.tsv'  # 替换为实际文件路径\n",
    "gtdb_df = pd.read_csv(gtdb_file,sep=\"\\t\")\n",
    "\n",
    "# 构建物种分类字典\n",
    "species_dict = dict(zip(gtdb_df['user_genome'], gtdb_df['classification']))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## ARG处理"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# DeepARG 输出文件夹路径\n",
    "sample_type=\"bio\"\n",
    "sample_type=\"nonbio\"\n",
    "# sample_type=\"water\"\n",
    "# sample_type=\"wood\"\n",
    "deeparg_dir = os.path.join('/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation',sample_type+'_bin','deeparg_bin_out')\n",
    "\n",
    "arg_metadata = []\n",
    "\n",
    "# 遍历deeparg文件夹中的每个文件\n",
    "for filename in os.listdir(deeparg_dir):\n",
    "    if filename.endswith('_DeepARG.out.mapping.ARG'):\n",
    "        bin_num = filename.split('_')[0]  # 提取bin编号\n",
    "        bin_name = f'bin.{bin_num.split(\".\")[1]}'\n",
    "        \n",
    "        # 读取ARG文件\n",
    "        arg_file = os.path.join(deeparg_dir, filename)\n",
    "        if os.path.getsize(arg_file) > 0:  # 处理非空文件\n",
    "            arg_df = pd.read_csv(arg_file, sep=\"\\t\",header=0)\n",
    "            for _, row in arg_df.iterrows():\n",
    "                entry_id = f\"{bin_name}_{row['read_id']}\"\n",
    "                contig_name =bin_num+\"_\"+entry_id.split('_')[1] + \"_\" + entry_id.split('_')[2]\n",
    "                arg_metadata.append({\n",
    "                    'id': entry_id,\n",
    "                    'bin_id':bin_name,\n",
    "                    'orf_name':row['read_id'],\n",
    "                    'species': species_dict.get(bin_name, 'Unknown'),\n",
    "                    'ARG': row['#ARG'],\n",
    "                    'predicted_ARG_class': row['predicted_ARG-class'],\n",
    "                    'contig_name': contig_name\n",
    "                })\n",
    "\n",
    "\n",
    "# 将ARG注释信息转换为DataFrame\n",
    "arg_df = pd.DataFrame(arg_metadata)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 毒理因子注释"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.85_vf.m8\n",
      "File bin.85_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.71_vf.m8\n",
      "File bin.71_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.7_vf.m8\n",
      "File bin.7_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.22_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.89_vf.m8\n",
      "File bin.89_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.87_vf.m8\n",
      "File bin.87_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.96_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.130_vf.m8\n",
      "File bin.130_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.114_vf.m8\n",
      "File bin.114_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.168_vf.m8\n",
      "File bin.168_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.88_vf.m8\n",
      "File bin.88_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.186_vf.m8\n",
      "File bin.186_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.2_vf.m8\n",
      "File bin.2_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.131_vf.m8\n",
      "File bin.131_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.93_vf.m8\n",
      "File bin.93_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.62_vf.m8\n",
      "File bin.62_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.79_vf.m8\n",
      "File bin.79_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.68_vf.m8\n",
      "File bin.68_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.119_vf.m8\n",
      "File bin.119_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.171_vf.m8\n",
      "File bin.171_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.92_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.16_vf.m8\n",
      "File bin.16_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.51_vf.m8\n",
      "File bin.51_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.39_vf.m8\n",
      "File bin.39_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.156_vf.m8\n",
      "File bin.156_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.12_vf.m8\n",
      "File bin.12_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.144_vf.m8\n",
      "File bin.144_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.178_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.193_vf.m8\n",
      "File bin.193_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.145_vf.m8\n",
      "File bin.145_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.60_vf.m8\n",
      "File bin.60_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.174_vf.m8\n",
      "File bin.174_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.59_vf.m8\n",
      "File bin.59_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.118_vf.m8\n",
      "File bin.118_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.91_vf.m8\n",
      "File bin.91_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.61_vf.m8\n",
      "File bin.61_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.52_vf.m8\n",
      "File bin.52_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.129_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.160_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.176_vf.m8\n",
      "File bin.176_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.38_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.83_vf.m8\n",
      "File bin.83_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.173_vf.m8\n",
      "File bin.173_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.1_vf.m8\n",
      "File bin.1_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.123_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.45_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.80_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.28_vf.m8\n",
      "File bin.28_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.41_vf.m8\n",
      "File bin.41_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.35_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.113_vf.m8\n",
      "File bin.113_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.192_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.188_vf.m8\n",
      "File bin.188_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.140_vf.m8\n",
      "File bin.140_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.33_vf.m8\n",
      "File bin.33_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.183_vf.m8\n",
      "File bin.183_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.99_vf.m8\n",
      "File bin.99_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.53_vf.m8\n",
      "File bin.53_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.84_vf.m8\n",
      "File bin.84_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.149_vf.m8\n",
      "File bin.149_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.67_vf.m8\n",
      "File bin.67_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.40_vf.m8\n",
      "File bin.40_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.112_vf.m8\n",
      "File bin.112_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.3_vf.m8\n",
      "File bin.3_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.36_vf.m8\n",
      "File bin.36_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.196_vf.m8\n",
      "File bin.196_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.182_vf.m8\n",
      "File bin.182_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.101_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.102_vf.m8\n",
      "File bin.102_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.111_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.139_vf.m8\n",
      "File bin.139_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.25_vf.m8\n",
      "File bin.25_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.19_vf.m8\n",
      "File bin.19_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.56_vf.m8\n",
      "File bin.56_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.175_vf.m8\n",
      "File bin.175_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.142_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.29_vf.m8\n",
      "File bin.29_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.4_vf.m8\n",
      "File bin.4_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.121_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.153_vf.m8\n",
      "File bin.153_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.109_vf.m8\n",
      "File bin.109_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.138_vf.m8\n",
      "File bin.138_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.159_vf.m8\n",
      "File bin.159_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.124_vf.m8\n",
      "File bin.124_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.127_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.165_vf.m8\n",
      "File bin.165_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.37_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.46_vf.m8\n",
      "File bin.46_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.105_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.167_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.181_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.42_vf.m8\n",
      "File bin.42_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.115_vf.m8\n",
      "File bin.115_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.164_vf.m8\n",
      "File bin.164_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.172_vf.m8\n",
      "File bin.172_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.9_vf.m8\n",
      "File bin.9_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.197_vf.m8\n",
      "File bin.197_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.125_vf.m8\n",
      "File bin.125_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.70_vf.m8\n",
      "File bin.70_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.135_vf.m8\n",
      "File bin.135_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.184_vf.m8\n",
      "File bin.184_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.136_vf.m8\n",
      "File bin.136_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.189_vf.m8\n",
      "File bin.189_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.166_vf.m8\n",
      "File bin.166_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.128_vf.m8\n",
      "File bin.128_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.137_vf.m8\n",
      "File bin.137_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.43_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.177_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.86_vf.m8\n",
      "File bin.86_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.58_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.82_vf.m8\n",
      "File bin.82_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.44_vf.m8\n",
      "File bin.44_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.152_vf.m8\n",
      "File bin.152_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.98_vf.m8\n",
      "File bin.98_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.103_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.107_vf.m8\n",
      "File bin.107_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.116_vf.m8\n",
      "File bin.116_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.49_vf.m8\n",
      "File bin.49_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.5_vf.m8\n",
      "File bin.5_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.23_vf.m8\n",
      "File bin.23_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.151_vf.m8\n",
      "File bin.151_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.94_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.110_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.81_vf.m8\n",
      "File bin.81_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.69_vf.m8\n",
      "File bin.69_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.27_vf.m8\n",
      "File bin.27_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.50_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.76_vf.m8\n",
      "File bin.76_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.47_vf.m8\n",
      "File bin.47_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.117_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.17_vf.m8\n",
      "File bin.17_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.21_vf.m8\n",
      "File bin.21_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.54_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.10_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.30_vf.m8\n",
      "File bin.30_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.34_vf.m8\n",
      "File bin.34_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.97_vf.m8\n",
      "File bin.97_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.64_vf.m8\n",
      "File bin.64_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.6_vf.m8\n",
      "File bin.6_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.20_vf.m8\n",
      "File bin.20_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.55_vf.m8\n",
      "File bin.55_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.75_vf.m8\n",
      "File bin.75_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.120_vf.m8\n",
      "File bin.120_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.132_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.147_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.122_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.77_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.65_vf.m8\n",
      "File bin.65_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.195_vf.m8\n",
      "File bin.195_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.161_vf.m8\n",
      "File bin.161_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.141_vf.m8\n",
      "File bin.141_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.134_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.108_vf.m8\n",
      "File bin.108_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.179_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.78_vf.m8\n",
      "File bin.78_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.158_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.150_vf.m8\n",
      "File bin.150_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.154_vf.m8\n",
      "File bin.154_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.170_vf.m8\n",
      "File bin.170_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.8_vf.m8\n",
      "File bin.8_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.106_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.90_vf.m8\n",
      "File bin.90_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.162_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.104_vf.m8\n",
      "File bin.104_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.163_vf.m8\n",
      "File bin.163_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.13_vf.m8\n",
      "File bin.13_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.95_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.100_vf.m8\n",
      "File bin.100_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.57_vf.m8\n",
      "File bin.57_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.73_vf.m8\n",
      "File bin.73_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.26_vf.m8\n",
      "File bin.26_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.133_vf.m8\n",
      "File bin.133_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.157_vf.m8\n",
      "File bin.157_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.146_vf.m8\n",
      "File bin.146_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.155_vf.m8\n",
      "File bin.155_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.148_vf.m8\n",
      "File bin.148_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.14_vf.m8\n",
      "File bin.14_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.31_vf.m8\n",
      "File bin.31_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.63_vf.m8\n",
      "File bin.63_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.18_vf.m8\n",
      "File bin.18_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.48_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.24_vf.m8\n",
      "File bin.24_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.169_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.32_vf.m8\n",
      "File bin.32_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.126_vf.m8\n",
      "File bin.126_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.15_vf.m8\n",
      "File bin.15_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.143_vf.m8\n",
      "File bin.143_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.11_vf.m8\n",
      "File bin.11_vf.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.74_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.72_vf.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/VFDB_mmseq_out/bin.180_vf.m8\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>bin_id</th>\n",
       "      <th>orf_name</th>\n",
       "      <th>sub_id</th>\n",
       "      <th>species</th>\n",
       "      <th>VF_NAME</th>\n",
       "      <th>contig_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>bin.22_k141_2000038_length_16138_cov_25.0723_8</td>\n",
       "      <td>bin.22</td>\n",
       "      <td>k141_2000038_length_16138_cov_25.0723_8</td>\n",
       "      <td>VFG014004(gb|WP_003148848)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.22_k141_2000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>bin.96_k141_7707864_length_9660_cov_12.2338_6</td>\n",
       "      <td>bin.96</td>\n",
       "      <td>k141_7707864_length_9660_cov_12.2338_6</td>\n",
       "      <td>VFG041187(gb|WP_012849267)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.96_k141_7707864</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>bin.92_k141_1962748_length_1612_cov_10.0843_2</td>\n",
       "      <td>bin.92</td>\n",
       "      <td>k141_1962748_length_1612_cov_10.0843_2</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.92_k141_1962748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>bin.178_k141_817152_length_30588_cov_23.4263_19</td>\n",
       "      <td>bin.178</td>\n",
       "      <td>k141_817152_length_30588_cov_23.4263_19</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.178_k141_817152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>bin.129_k141_5151809_length_283910_cov_32.2217...</td>\n",
       "      <td>bin.129</td>\n",
       "      <td>k141_5151809_length_283910_cov_32.2217_102</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.129_k141_5151809</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>bin.160_k141_7916819_length_12396_cov_25.3947_3</td>\n",
       "      <td>bin.160</td>\n",
       "      <td>k141_7916819_length_12396_cov_25.3947_3</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.160_k141_7916819</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>bin.38_k141_3482740_length_5717_cov_9.0000_1</td>\n",
       "      <td>bin.38</td>\n",
       "      <td>k141_3482740_length_5717_cov_9.0000_1</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.38_k141_3482740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>bin.38_k141_987689_length_10950_cov_10.0000_7</td>\n",
       "      <td>bin.38</td>\n",
       "      <td>k141_987689_length_10950_cov_10.0000_7</td>\n",
       "      <td>VFG011429(gb|WP_002963985)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.38_k141_987689</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>bin.123_k141_1899457_length_23673_cov_13.6425_15</td>\n",
       "      <td>bin.123</td>\n",
       "      <td>k141_1899457_length_23673_cov_13.6425_15</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.123_k141_1899457</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>bin.45_k141_4774825_length_2608_cov_10.1536_2</td>\n",
       "      <td>bin.45</td>\n",
       "      <td>k141_4774825_length_2608_cov_10.1536_2</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.45_k141_4774825</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>bin.80_k141_5121601_length_23456_cov_13.4137_4</td>\n",
       "      <td>bin.80</td>\n",
       "      <td>k141_5121601_length_23456_cov_13.4137_4</td>\n",
       "      <td>VFG042942(gb|WP_004085459)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.80_k141_5121601</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>bin.80_k141_5798227_length_3175_cov_13.0046_2</td>\n",
       "      <td>bin.80</td>\n",
       "      <td>k141_5798227_length_3175_cov_13.0046_2</td>\n",
       "      <td>VFG042945(gb|WP_004084948)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.80_k141_5798227</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>bin.80_k141_779923_length_23168_cov_12.6355_21</td>\n",
       "      <td>bin.80</td>\n",
       "      <td>k141_779923_length_23168_cov_12.6355_21</td>\n",
       "      <td>VFG046459(gb|WP_004287053)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.80_k141_779923</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>bin.35_k141_2263980_length_24743_cov_29.9409_11</td>\n",
       "      <td>bin.35</td>\n",
       "      <td>k141_2263980_length_24743_cov_29.9409_11</td>\n",
       "      <td>VFG013907(gb|WP_004373763)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.35_k141_2263980</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>bin.35_k141_2872763_length_24962_cov_22.8465_2</td>\n",
       "      <td>bin.35</td>\n",
       "      <td>k141_2872763_length_24962_cov_22.8465_2</td>\n",
       "      <td>VFG013954(gb|WP_003245164)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.35_k141_2872763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>bin.35_k141_652770_length_10033_cov_17.6450_6</td>\n",
       "      <td>bin.35</td>\n",
       "      <td>k141_652770_length_10033_cov_17.6450_6</td>\n",
       "      <td>VFG045727(gb|WP_003636647)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.35_k141_652770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>bin.192_k141_10063824_length_1839_cov_4.0000_3</td>\n",
       "      <td>bin.192</td>\n",
       "      <td>k141_10063824_length_1839_cov_4.0000_3</td>\n",
       "      <td>VFG041187(gb|WP_012849267)</td>\n",
       "      <td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.192_k141_10063824</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>bin.101_k141_1251047_length_6777_cov_77.0750_3</td>\n",
       "      <td>bin.101</td>\n",
       "      <td>k141_1251047_length_6777_cov_77.0750_3</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.101_k141_1251047</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>bin.111_k141_6527414_length_2849_cov_12.4103_2</td>\n",
       "      <td>bin.111</td>\n",
       "      <td>k141_6527414_length_2849_cov_12.4103_2</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.111_k141_6527414</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>bin.142_k141_4899566_length_18712_cov_24.7942_11</td>\n",
       "      <td>bin.142</td>\n",
       "      <td>k141_4899566_length_18712_cov_24.7942_11</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.142_k141_4899566</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>bin.121_k141_1457360_length_41757_cov_116.2524_24</td>\n",
       "      <td>bin.121</td>\n",
       "      <td>k141_1457360_length_41757_cov_116.2524_24</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.121_k141_1457360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>bin.127_k141_4452299_length_8591_cov_12.0000_9</td>\n",
       "      <td>bin.127</td>\n",
       "      <td>k141_4452299_length_8591_cov_12.0000_9</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.127_k141_4452299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>bin.37_k141_8000741_length_2657_cov_6.0000_2</td>\n",
       "      <td>bin.37</td>\n",
       "      <td>k141_8000741_length_2657_cov_6.0000_2</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.37_k141_8000741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>bin.105_k141_2446063_length_4363_cov_18.3221_2</td>\n",
       "      <td>bin.105</td>\n",
       "      <td>k141_2446063_length_4363_cov_18.3221_2</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.105_k141_2446063</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>bin.167_k141_8102349_length_3114_cov_21.2842_2</td>\n",
       "      <td>bin.167</td>\n",
       "      <td>k141_8102349_length_3114_cov_21.2842_2</td>\n",
       "      <td>VFG042945(gb|WP_004084948)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.167_k141_8102349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>bin.167_k141_8127610_length_14229_cov_32.9873_5</td>\n",
       "      <td>bin.167</td>\n",
       "      <td>k141_8127610_length_14229_cov_32.9873_5</td>\n",
       "      <td>VFG042942(gb|WP_004085459)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.167_k141_8127610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>bin.181_k141_5405823_length_13674_cov_17.3253_14</td>\n",
       "      <td>bin.181</td>\n",
       "      <td>k141_5405823_length_13674_cov_17.3253_14</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.181_k141_5405823</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>bin.43_k141_10018952_length_38267_cov_11.0009_34</td>\n",
       "      <td>bin.43</td>\n",
       "      <td>k141_10018952_length_38267_cov_11.0009_34</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.43_k141_10018952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>bin.177_k141_10168146_length_15903_cov_8.0000_17</td>\n",
       "      <td>bin.177</td>\n",
       "      <td>k141_10168146_length_15903_cov_8.0000_17</td>\n",
       "      <td>VFG011626(gb|WP_002965156)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>BvrR-BvrS</td>\n",
       "      <td>bin.177_k141_10168146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>bin.177_k141_8426065_length_4161_cov_7.6082_6</td>\n",
       "      <td>bin.177</td>\n",
       "      <td>k141_8426065_length_4161_cov_7.6082_6</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.177_k141_8426065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>bin.58_k141_3256981_length_36512_cov_16.2791_37</td>\n",
       "      <td>bin.58</td>\n",
       "      <td>k141_3256981_length_36512_cov_16.2791_37</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.58_k141_3256981</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>bin.103_k141_8440949_length_6504_cov_14.9206_4</td>\n",
       "      <td>bin.103</td>\n",
       "      <td>k141_8440949_length_6504_cov_14.9206_4</td>\n",
       "      <td>VFG020006(gb|WP_010207547)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.103_k141_8440949</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>bin.94_k141_5932482_length_8915_cov_18.7946_6</td>\n",
       "      <td>bin.94</td>\n",
       "      <td>k141_5932482_length_8915_cov_18.7946_6</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.94_k141_5932482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>bin.110_k141_1807550_length_18918_cov_21.0703_12</td>\n",
       "      <td>bin.110</td>\n",
       "      <td>k141_1807550_length_18918_cov_21.0703_12</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.110_k141_1807550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>bin.50_k141_924461_length_28402_cov_12.1886_24</td>\n",
       "      <td>bin.50</td>\n",
       "      <td>k141_924461_length_28402_cov_12.1886_24</td>\n",
       "      <td>VFG045727(gb|WP_003636647)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.50_k141_924461</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>bin.50_k141_9485022_length_11201_cov_13.1224_1</td>\n",
       "      <td>bin.50</td>\n",
       "      <td>k141_9485022_length_11201_cov_13.1224_1</td>\n",
       "      <td>VFG046474(gb|WP_014714676)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.50_k141_9485022</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>bin.117_k141_10384956_length_18329_cov_14.7077_16</td>\n",
       "      <td>bin.117</td>\n",
       "      <td>k141_10384956_length_18329_cov_14.7077_16</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.117_k141_10384956</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>bin.117_k141_4971356_length_11981_cov_12.4992_2</td>\n",
       "      <td>bin.117</td>\n",
       "      <td>k141_4971356_length_11981_cov_12.4992_2</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.117_k141_4971356</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>bin.54_k141_4260039_length_9621_cov_16.6333_12</td>\n",
       "      <td>bin.54</td>\n",
       "      <td>k141_4260039_length_9621_cov_16.6333_12</td>\n",
       "      <td>VFG013907(gb|WP_004373763)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.54_k141_4260039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>bin.10_k141_7363498_length_4798_cov_12.0000_3</td>\n",
       "      <td>bin.10</td>\n",
       "      <td>k141_7363498_length_4798_cov_12.0000_3</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.10_k141_7363498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>bin.132_k141_6289086_length_41258_cov_14.1064_5</td>\n",
       "      <td>bin.132</td>\n",
       "      <td>k141_6289086_length_41258_cov_14.1064_5</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.132_k141_6289086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>bin.147_k141_1660470_length_4943_cov_181.7349_5</td>\n",
       "      <td>bin.147</td>\n",
       "      <td>k141_1660470_length_4943_cov_181.7349_5</td>\n",
       "      <td>VFG042945(gb|WP_004084948)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.147_k141_1660470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>bin.147_k141_1968028_length_5670_cov_218.0610_2</td>\n",
       "      <td>bin.147</td>\n",
       "      <td>k141_1968028_length_5670_cov_218.0610_2</td>\n",
       "      <td>VFG007573(gb|WP_005420316)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.147_k141_1968028</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>bin.147_k141_2983909_length_2482_cov_188.8223_4</td>\n",
       "      <td>bin.147</td>\n",
       "      <td>k141_2983909_length_2482_cov_188.8223_4</td>\n",
       "      <td>VFG042942(gb|WP_004085459)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.147_k141_2983909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>bin.122_k141_10227074_length_3412_cov_7.0018_2</td>\n",
       "      <td>bin.122</td>\n",
       "      <td>k141_10227074_length_3412_cov_7.0018_2</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.122_k141_10227074</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>bin.77_k141_854285_length_6413_cov_18.2121_2</td>\n",
       "      <td>bin.77</td>\n",
       "      <td>k141_854285_length_6413_cov_18.2121_2</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.77_k141_854285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>bin.134_k141_2302151_length_5104_cov_9.0000_6</td>\n",
       "      <td>bin.134</td>\n",
       "      <td>k141_2302151_length_5104_cov_9.0000_6</td>\n",
       "      <td>VFG013907(gb|WP_004373763)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.134_k141_2302151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>bin.179_k141_4072224_length_10372_cov_11.0000_10</td>\n",
       "      <td>bin.179</td>\n",
       "      <td>k141_4072224_length_10372_cov_11.0000_10</td>\n",
       "      <td>VFG041187(gb|WP_012849267)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.179_k141_4072224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>bin.158_k141_6228527_length_2604_cov_21.2574_2</td>\n",
       "      <td>bin.158</td>\n",
       "      <td>k141_6228527_length_2604_cov_21.2574_2</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.158_k141_6228527</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>bin.106_k141_498486_length_10306_cov_20.3603_12</td>\n",
       "      <td>bin.106</td>\n",
       "      <td>k141_498486_length_10306_cov_20.3603_12</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.106_k141_498486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>bin.162_k141_2509948_length_2134_cov_8.0070_2</td>\n",
       "      <td>bin.162</td>\n",
       "      <td>k141_2509948_length_2134_cov_8.0070_2</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.162_k141_2509948</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>bin.95_k141_566354_length_23627_cov_26.0687_6</td>\n",
       "      <td>bin.95</td>\n",
       "      <td>k141_566354_length_23627_cov_26.0687_6</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.95_k141_566354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>bin.48_k141_3572535_length_5201_cov_31.9433_5</td>\n",
       "      <td>bin.48</td>\n",
       "      <td>k141_3572535_length_5201_cov_31.9433_5</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.48_k141_3572535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>bin.169_k141_4119434_length_42635_cov_56.4316_35</td>\n",
       "      <td>bin.169</td>\n",
       "      <td>k141_4119434_length_42635_cov_56.4316_35</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.169_k141_4119434</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>bin.74_k141_4181161_length_3284_cov_9.2504_5</td>\n",
       "      <td>bin.74</td>\n",
       "      <td>k141_4181161_length_3284_cov_9.2504_5</td>\n",
       "      <td>VFG013907(gb|WP_004373763)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.74_k141_4181161</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>bin.72_k141_433199_length_4806_cov_17.1614_2</td>\n",
       "      <td>bin.72</td>\n",
       "      <td>k141_433199_length_4806_cov_17.1614_2</td>\n",
       "      <td>VFG011430(gb|WP_002963616)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>LPS</td>\n",
       "      <td>bin.72_k141_433199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>bin.180_k141_3384492_length_19785_cov_9.0000_12</td>\n",
       "      <td>bin.180</td>\n",
       "      <td>k141_3384492_length_19785_cov_9.0000_12</td>\n",
       "      <td>VFG046459(gb|WP_004287053)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.180_k141_3384492</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>bin.180_k141_9345294_length_2445_cov_8.0000_1</td>\n",
       "      <td>bin.180</td>\n",
       "      <td>k141_9345294_length_2445_cov_8.0000_1</td>\n",
       "      <td>VFG046474(gb|WP_014714676)</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.180_k141_9345294</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   id   bin_id  \\\n",
       "0      bin.22_k141_2000038_length_16138_cov_25.0723_8   bin.22   \n",
       "1       bin.96_k141_7707864_length_9660_cov_12.2338_6   bin.96   \n",
       "2       bin.92_k141_1962748_length_1612_cov_10.0843_2   bin.92   \n",
       "3     bin.178_k141_817152_length_30588_cov_23.4263_19  bin.178   \n",
       "4   bin.129_k141_5151809_length_283910_cov_32.2217...  bin.129   \n",
       "5     bin.160_k141_7916819_length_12396_cov_25.3947_3  bin.160   \n",
       "6        bin.38_k141_3482740_length_5717_cov_9.0000_1   bin.38   \n",
       "7       bin.38_k141_987689_length_10950_cov_10.0000_7   bin.38   \n",
       "8    bin.123_k141_1899457_length_23673_cov_13.6425_15  bin.123   \n",
       "9       bin.45_k141_4774825_length_2608_cov_10.1536_2   bin.45   \n",
       "10     bin.80_k141_5121601_length_23456_cov_13.4137_4   bin.80   \n",
       "11      bin.80_k141_5798227_length_3175_cov_13.0046_2   bin.80   \n",
       "12     bin.80_k141_779923_length_23168_cov_12.6355_21   bin.80   \n",
       "13    bin.35_k141_2263980_length_24743_cov_29.9409_11   bin.35   \n",
       "14     bin.35_k141_2872763_length_24962_cov_22.8465_2   bin.35   \n",
       "15      bin.35_k141_652770_length_10033_cov_17.6450_6   bin.35   \n",
       "16     bin.192_k141_10063824_length_1839_cov_4.0000_3  bin.192   \n",
       "17     bin.101_k141_1251047_length_6777_cov_77.0750_3  bin.101   \n",
       "18     bin.111_k141_6527414_length_2849_cov_12.4103_2  bin.111   \n",
       "19   bin.142_k141_4899566_length_18712_cov_24.7942_11  bin.142   \n",
       "20  bin.121_k141_1457360_length_41757_cov_116.2524_24  bin.121   \n",
       "21     bin.127_k141_4452299_length_8591_cov_12.0000_9  bin.127   \n",
       "22       bin.37_k141_8000741_length_2657_cov_6.0000_2   bin.37   \n",
       "23     bin.105_k141_2446063_length_4363_cov_18.3221_2  bin.105   \n",
       "24     bin.167_k141_8102349_length_3114_cov_21.2842_2  bin.167   \n",
       "25    bin.167_k141_8127610_length_14229_cov_32.9873_5  bin.167   \n",
       "26   bin.181_k141_5405823_length_13674_cov_17.3253_14  bin.181   \n",
       "27   bin.43_k141_10018952_length_38267_cov_11.0009_34   bin.43   \n",
       "28   bin.177_k141_10168146_length_15903_cov_8.0000_17  bin.177   \n",
       "29      bin.177_k141_8426065_length_4161_cov_7.6082_6  bin.177   \n",
       "30    bin.58_k141_3256981_length_36512_cov_16.2791_37   bin.58   \n",
       "31     bin.103_k141_8440949_length_6504_cov_14.9206_4  bin.103   \n",
       "32      bin.94_k141_5932482_length_8915_cov_18.7946_6   bin.94   \n",
       "33   bin.110_k141_1807550_length_18918_cov_21.0703_12  bin.110   \n",
       "34     bin.50_k141_924461_length_28402_cov_12.1886_24   bin.50   \n",
       "35     bin.50_k141_9485022_length_11201_cov_13.1224_1   bin.50   \n",
       "36  bin.117_k141_10384956_length_18329_cov_14.7077_16  bin.117   \n",
       "37    bin.117_k141_4971356_length_11981_cov_12.4992_2  bin.117   \n",
       "38     bin.54_k141_4260039_length_9621_cov_16.6333_12   bin.54   \n",
       "39      bin.10_k141_7363498_length_4798_cov_12.0000_3   bin.10   \n",
       "40    bin.132_k141_6289086_length_41258_cov_14.1064_5  bin.132   \n",
       "41    bin.147_k141_1660470_length_4943_cov_181.7349_5  bin.147   \n",
       "42    bin.147_k141_1968028_length_5670_cov_218.0610_2  bin.147   \n",
       "43    bin.147_k141_2983909_length_2482_cov_188.8223_4  bin.147   \n",
       "44     bin.122_k141_10227074_length_3412_cov_7.0018_2  bin.122   \n",
       "45       bin.77_k141_854285_length_6413_cov_18.2121_2   bin.77   \n",
       "46      bin.134_k141_2302151_length_5104_cov_9.0000_6  bin.134   \n",
       "47   bin.179_k141_4072224_length_10372_cov_11.0000_10  bin.179   \n",
       "48     bin.158_k141_6228527_length_2604_cov_21.2574_2  bin.158   \n",
       "49    bin.106_k141_498486_length_10306_cov_20.3603_12  bin.106   \n",
       "50      bin.162_k141_2509948_length_2134_cov_8.0070_2  bin.162   \n",
       "51      bin.95_k141_566354_length_23627_cov_26.0687_6   bin.95   \n",
       "52      bin.48_k141_3572535_length_5201_cov_31.9433_5   bin.48   \n",
       "53   bin.169_k141_4119434_length_42635_cov_56.4316_35  bin.169   \n",
       "54       bin.74_k141_4181161_length_3284_cov_9.2504_5   bin.74   \n",
       "55       bin.72_k141_433199_length_4806_cov_17.1614_2   bin.72   \n",
       "56    bin.180_k141_3384492_length_19785_cov_9.0000_12  bin.180   \n",
       "57      bin.180_k141_9345294_length_2445_cov_8.0000_1  bin.180   \n",
       "\n",
       "                                      orf_name                      sub_id  \\\n",
       "0      k141_2000038_length_16138_cov_25.0723_8  VFG014004(gb|WP_003148848)   \n",
       "1       k141_7707864_length_9660_cov_12.2338_6  VFG041187(gb|WP_012849267)   \n",
       "2       k141_1962748_length_1612_cov_10.0843_2  VFG011430(gb|WP_002963616)   \n",
       "3      k141_817152_length_30588_cov_23.4263_19  VFG011430(gb|WP_002963616)   \n",
       "4   k141_5151809_length_283910_cov_32.2217_102  VFG011430(gb|WP_002963616)   \n",
       "5      k141_7916819_length_12396_cov_25.3947_3  VFG011430(gb|WP_002963616)   \n",
       "6        k141_3482740_length_5717_cov_9.0000_1  VFG011430(gb|WP_002963616)   \n",
       "7       k141_987689_length_10950_cov_10.0000_7  VFG011429(gb|WP_002963985)   \n",
       "8     k141_1899457_length_23673_cov_13.6425_15  VFG011430(gb|WP_002963616)   \n",
       "9       k141_4774825_length_2608_cov_10.1536_2  VFG011430(gb|WP_002963616)   \n",
       "10     k141_5121601_length_23456_cov_13.4137_4  VFG042942(gb|WP_004085459)   \n",
       "11      k141_5798227_length_3175_cov_13.0046_2  VFG042945(gb|WP_004084948)   \n",
       "12     k141_779923_length_23168_cov_12.6355_21  VFG046459(gb|WP_004287053)   \n",
       "13    k141_2263980_length_24743_cov_29.9409_11  VFG013907(gb|WP_004373763)   \n",
       "14     k141_2872763_length_24962_cov_22.8465_2  VFG013954(gb|WP_003245164)   \n",
       "15      k141_652770_length_10033_cov_17.6450_6  VFG045727(gb|WP_003636647)   \n",
       "16      k141_10063824_length_1839_cov_4.0000_3  VFG041187(gb|WP_012849267)   \n",
       "17      k141_1251047_length_6777_cov_77.0750_3  VFG011430(gb|WP_002963616)   \n",
       "18      k141_6527414_length_2849_cov_12.4103_2  VFG011430(gb|WP_002963616)   \n",
       "19    k141_4899566_length_18712_cov_24.7942_11  VFG011430(gb|WP_002963616)   \n",
       "20   k141_1457360_length_41757_cov_116.2524_24  VFG011430(gb|WP_002963616)   \n",
       "21      k141_4452299_length_8591_cov_12.0000_9  VFG011430(gb|WP_002963616)   \n",
       "22       k141_8000741_length_2657_cov_6.0000_2  VFG011430(gb|WP_002963616)   \n",
       "23      k141_2446063_length_4363_cov_18.3221_2  VFG011430(gb|WP_002963616)   \n",
       "24      k141_8102349_length_3114_cov_21.2842_2  VFG042945(gb|WP_004084948)   \n",
       "25     k141_8127610_length_14229_cov_32.9873_5  VFG042942(gb|WP_004085459)   \n",
       "26    k141_5405823_length_13674_cov_17.3253_14  VFG011430(gb|WP_002963616)   \n",
       "27   k141_10018952_length_38267_cov_11.0009_34  VFG011430(gb|WP_002963616)   \n",
       "28    k141_10168146_length_15903_cov_8.0000_17  VFG011626(gb|WP_002965156)   \n",
       "29       k141_8426065_length_4161_cov_7.6082_6  VFG011430(gb|WP_002963616)   \n",
       "30    k141_3256981_length_36512_cov_16.2791_37  VFG011430(gb|WP_002963616)   \n",
       "31      k141_8440949_length_6504_cov_14.9206_4  VFG020006(gb|WP_010207547)   \n",
       "32      k141_5932482_length_8915_cov_18.7946_6  VFG011430(gb|WP_002963616)   \n",
       "33    k141_1807550_length_18918_cov_21.0703_12  VFG011430(gb|WP_002963616)   \n",
       "34     k141_924461_length_28402_cov_12.1886_24  VFG045727(gb|WP_003636647)   \n",
       "35     k141_9485022_length_11201_cov_13.1224_1  VFG046474(gb|WP_014714676)   \n",
       "36   k141_10384956_length_18329_cov_14.7077_16  VFG011430(gb|WP_002963616)   \n",
       "37     k141_4971356_length_11981_cov_12.4992_2  VFG011430(gb|WP_002963616)   \n",
       "38     k141_4260039_length_9621_cov_16.6333_12  VFG013907(gb|WP_004373763)   \n",
       "39      k141_7363498_length_4798_cov_12.0000_3  VFG011430(gb|WP_002963616)   \n",
       "40     k141_6289086_length_41258_cov_14.1064_5  VFG011430(gb|WP_002963616)   \n",
       "41     k141_1660470_length_4943_cov_181.7349_5  VFG042945(gb|WP_004084948)   \n",
       "42     k141_1968028_length_5670_cov_218.0610_2  VFG007573(gb|WP_005420316)   \n",
       "43     k141_2983909_length_2482_cov_188.8223_4  VFG042942(gb|WP_004085459)   \n",
       "44      k141_10227074_length_3412_cov_7.0018_2  VFG011430(gb|WP_002963616)   \n",
       "45       k141_854285_length_6413_cov_18.2121_2  VFG011430(gb|WP_002963616)   \n",
       "46       k141_2302151_length_5104_cov_9.0000_6  VFG013907(gb|WP_004373763)   \n",
       "47    k141_4072224_length_10372_cov_11.0000_10  VFG041187(gb|WP_012849267)   \n",
       "48      k141_6228527_length_2604_cov_21.2574_2  VFG011430(gb|WP_002963616)   \n",
       "49     k141_498486_length_10306_cov_20.3603_12  VFG011430(gb|WP_002963616)   \n",
       "50       k141_2509948_length_2134_cov_8.0070_2  VFG011430(gb|WP_002963616)   \n",
       "51      k141_566354_length_23627_cov_26.0687_6  VFG011430(gb|WP_002963616)   \n",
       "52      k141_3572535_length_5201_cov_31.9433_5  VFG011430(gb|WP_002963616)   \n",
       "53    k141_4119434_length_42635_cov_56.4316_35  VFG011430(gb|WP_002963616)   \n",
       "54       k141_4181161_length_3284_cov_9.2504_5  VFG013907(gb|WP_004373763)   \n",
       "55       k141_433199_length_4806_cov_17.1614_2  VFG011430(gb|WP_002963616)   \n",
       "56     k141_3384492_length_19785_cov_9.0000_12  VFG046459(gb|WP_004287053)   \n",
       "57       k141_9345294_length_2445_cov_8.0000_1  VFG046474(gb|WP_014714676)   \n",
       "\n",
       "                                              species    VF_NAME  \\\n",
       "0   d__Bacteria;p__Proteobacteria;c__Gammaproteoba...          -   \n",
       "1   d__Bacteria;p__Proteobacteria;c__Gammaproteoba...          -   \n",
       "2   d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "3   d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "4   d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "5   d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "6   d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "7   d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "8   d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "9   d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "10  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...          -   \n",
       "11  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...          -   \n",
       "12  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...          -   \n",
       "13  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...          -   \n",
       "14  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...          -   \n",
       "15  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...          -   \n",
       "16  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...          -   \n",
       "17  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "18  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "19  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "20  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "21  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "22  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "23  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "24  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...          -   \n",
       "25  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...          -   \n",
       "26  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "27  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "28  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...  BvrR-BvrS   \n",
       "29  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "30  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "31  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...          -   \n",
       "32  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "33  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "34  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...          -   \n",
       "35  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...          -   \n",
       "36  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "37  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "38  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...          -   \n",
       "39  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "40  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "41  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...          -   \n",
       "42  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...          -   \n",
       "43  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...          -   \n",
       "44  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "45  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "46  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...          -   \n",
       "47  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...          -   \n",
       "48  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "49  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "50  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "51  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "52  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "53  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "54  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...          -   \n",
       "55  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        LPS   \n",
       "56  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...          -   \n",
       "57  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...          -   \n",
       "\n",
       "              contig_name  \n",
       "0     bin.22_k141_2000038  \n",
       "1     bin.96_k141_7707864  \n",
       "2     bin.92_k141_1962748  \n",
       "3     bin.178_k141_817152  \n",
       "4    bin.129_k141_5151809  \n",
       "5    bin.160_k141_7916819  \n",
       "6     bin.38_k141_3482740  \n",
       "7      bin.38_k141_987689  \n",
       "8    bin.123_k141_1899457  \n",
       "9     bin.45_k141_4774825  \n",
       "10    bin.80_k141_5121601  \n",
       "11    bin.80_k141_5798227  \n",
       "12     bin.80_k141_779923  \n",
       "13    bin.35_k141_2263980  \n",
       "14    bin.35_k141_2872763  \n",
       "15     bin.35_k141_652770  \n",
       "16  bin.192_k141_10063824  \n",
       "17   bin.101_k141_1251047  \n",
       "18   bin.111_k141_6527414  \n",
       "19   bin.142_k141_4899566  \n",
       "20   bin.121_k141_1457360  \n",
       "21   bin.127_k141_4452299  \n",
       "22    bin.37_k141_8000741  \n",
       "23   bin.105_k141_2446063  \n",
       "24   bin.167_k141_8102349  \n",
       "25   bin.167_k141_8127610  \n",
       "26   bin.181_k141_5405823  \n",
       "27   bin.43_k141_10018952  \n",
       "28  bin.177_k141_10168146  \n",
       "29   bin.177_k141_8426065  \n",
       "30    bin.58_k141_3256981  \n",
       "31   bin.103_k141_8440949  \n",
       "32    bin.94_k141_5932482  \n",
       "33   bin.110_k141_1807550  \n",
       "34     bin.50_k141_924461  \n",
       "35    bin.50_k141_9485022  \n",
       "36  bin.117_k141_10384956  \n",
       "37   bin.117_k141_4971356  \n",
       "38    bin.54_k141_4260039  \n",
       "39    bin.10_k141_7363498  \n",
       "40   bin.132_k141_6289086  \n",
       "41   bin.147_k141_1660470  \n",
       "42   bin.147_k141_1968028  \n",
       "43   bin.147_k141_2983909  \n",
       "44  bin.122_k141_10227074  \n",
       "45     bin.77_k141_854285  \n",
       "46   bin.134_k141_2302151  \n",
       "47   bin.179_k141_4072224  \n",
       "48   bin.158_k141_6228527  \n",
       "49    bin.106_k141_498486  \n",
       "50   bin.162_k141_2509948  \n",
       "51     bin.95_k141_566354  \n",
       "52    bin.48_k141_3572535  \n",
       "53   bin.169_k141_4119434  \n",
       "54    bin.74_k141_4181161  \n",
       "55     bin.72_k141_433199  \n",
       "56   bin.180_k141_3384492  \n",
       "57   bin.180_k141_9345294  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# VFDB metadata文件\n",
    "\n",
    "vfdb_metadata_file = '/storage/jufengLab/luogaoyang/db/VFDB/CompRanking_Virulence_Summary.tsv'\n",
    "vfdb_df = pd.read_csv(vfdb_metadata_file,sep=\"\\t\",index_col=0)\n",
    "vfdb_df\n",
    "# 构建VF字典\n",
    "vf_dict = dict(zip(vfdb_df['fasID'], vfdb_df['VF_Name']))\n",
    "\n",
    "# 毒力因子注释目录\n",
    "vf_dir = os.path.join('/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation',sample_type+'_bin','VFDB_mmseq_out')  # 替换为实际路径\n",
    "\n",
    "vf_metadata = []\n",
    "\n",
    "# 遍历毒力因子的注释文件\n",
    "for vf_file in os.listdir(vf_dir):\n",
    "    if vf_file.endswith('.m8'):\n",
    "        bin_num = vf_file.split('_')[0]\n",
    "        vf_file_path = os.path.join(vf_dir, vf_file)\n",
    "        print(vf_file_path)\n",
    "         # 检查文件是否为空\n",
    "        if os.path.getsize(vf_file_path) == 0:\n",
    "            print(f\"File {vf_file} is empty, skipping.\")\n",
    "            continue\n",
    "        \n",
    "        # 读取毒力因子注释文件\n",
    "        try:\n",
    "            df = pd.read_csv(vf_file_path, sep=\"\\t\", header=None)\n",
    "        except pd.errors.EmptyDataError:\n",
    "            print(f\"File {vf_file} is empty or cannot be parsed, skipping.\")\n",
    "            continue\n",
    "        \n",
    "        df.columns = ['id', 'sub_id', 'identity', 'alignLen', 'mismat', 'gapOpens', 'qStart', 'qEnd', 'sStart', 'sEnd', 'eval', 'bit']\n",
    "        \n",
    "        # 过滤出符合条件的毒力因子注释\n",
    "        filter_df = df[(df['identity'] > 0.6) & (df['alignLen'] > 40)]\n",
    "        # 去重操作：对于每个 id，选择 identity 最大的行\n",
    "        filter_df = filter_df.loc[filter_df.groupby('id')['identity'].idxmax()]\n",
    "        \n",
    "        for _, row in filter_df.iterrows():\n",
    "            entry_id = f\"{bin_num}_{row['id']}\"\n",
    "            contig_name =bin_num+\"_\"+entry_id.split('_')[1] + \"_\" + entry_id.split('_')[2]\n",
    "            vf_metadata.append({\n",
    "                'id': entry_id,\n",
    "                'bin_id':bin_num,\n",
    "                'orf_name':row['id'],\n",
    "                'sub_id': row['sub_id'],\n",
    "                'species': species_dict.get(bin_num, 'Unknown'),\n",
    "                'VF_NAME': vf_dict.get(row['sub_id'], '-'),\n",
    "                'contig_name': contig_name\n",
    "            })\n",
    "\n",
    "# 将毒力因子注释信息转换为DataFrame\n",
    "vf_df = pd.DataFrame(vf_metadata)\n",
    "vf_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 塑料降解基因"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.153_plsdb.m8\n",
      "File bin.153_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.90_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.93_plsdb.m8\n",
      "File bin.93_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.116_plsdb.m8\n",
      "File bin.116_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.62_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.175_plsdb.m8\n",
      "File bin.175_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.110_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.154_plsdb.m8\n",
      "File bin.154_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.135_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.136_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.123_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.184_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.115_plsdb.m8\n",
      "File bin.115_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.159_plsdb.m8\n",
      "File bin.159_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.83_plsdb.m8\n",
      "File bin.83_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.97_plsdb.m8\n",
      "File bin.97_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.181_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.119_plsdb.m8\n",
      "File bin.119_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.43_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.1_plsdb.m8\n",
      "File bin.1_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.81_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.6_plsdb.m8\n",
      "File bin.6_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.48_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.179_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.84_plsdb.m8\n",
      "File bin.84_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.152_plsdb.m8\n",
      "File bin.152_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.78_plsdb.m8\n",
      "File bin.78_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.99_plsdb.m8\n",
      "File bin.99_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.122_plsdb.m8\n",
      "File bin.122_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.91_plsdb.m8\n",
      "File bin.91_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.176_plsdb.m8\n",
      "File bin.176_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.54_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.147_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.166_plsdb.m8\n",
      "File bin.166_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.40_plsdb.m8\n",
      "File bin.40_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.70_plsdb.m8\n",
      "File bin.70_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.171_plsdb.m8\n",
      "File bin.171_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.85_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.33_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.29_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.129_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.148_plsdb.m8\n",
      "File bin.148_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.55_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.92_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.101_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.89_plsdb.m8\n",
      "File bin.89_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.59_plsdb.m8\n",
      "File bin.59_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.8_plsdb.m8\n",
      "File bin.8_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.42_plsdb.m8\n",
      "File bin.42_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.80_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.146_plsdb.m8\n",
      "File bin.146_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.150_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.196_plsdb.m8\n",
      "File bin.196_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.182_plsdb.m8\n",
      "File bin.182_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.102_plsdb.m8\n",
      "File bin.102_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.87_plsdb.m8\n",
      "File bin.87_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.113_plsdb.m8\n",
      "File bin.113_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.197_plsdb.m8\n",
      "File bin.197_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.46_plsdb.m8\n",
      "File bin.46_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.31_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.134_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.52_plsdb.m8\n",
      "File bin.52_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.138_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.104_plsdb.m8\n",
      "File bin.104_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.86_plsdb.m8\n",
      "File bin.86_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.14_plsdb.m8\n",
      "File bin.14_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.131_plsdb.m8\n",
      "File bin.131_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.95_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.108_plsdb.m8\n",
      "File bin.108_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.163_plsdb.m8\n",
      "File bin.163_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.68_plsdb.m8\n",
      "File bin.68_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.107_plsdb.m8\n",
      "File bin.107_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.49_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.57_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.13_plsdb.m8\n",
      "File bin.13_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.94_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.51_plsdb.m8\n",
      "File bin.51_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.20_plsdb.m8\n",
      "File bin.20_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.192_plsdb.m8\n",
      "File bin.192_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.126_plsdb.m8\n",
      "File bin.126_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.74_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.130_plsdb.m8\n",
      "File bin.130_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.71_plsdb.m8\n",
      "File bin.71_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.120_plsdb.m8\n",
      "File bin.120_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.79_plsdb.m8\n",
      "File bin.79_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.168_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.39_plsdb.m8\n",
      "File bin.39_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.38_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.193_plsdb.m8\n",
      "File bin.193_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.2_plsdb.m8\n",
      "File bin.2_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.160_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.64_plsdb.m8\n",
      "File bin.64_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.28_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.75_plsdb.m8\n",
      "File bin.75_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.72_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.112_plsdb.m8\n",
      "File bin.112_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.121_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.103_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.76_plsdb.m8\n",
      "File bin.76_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.60_plsdb.m8\n",
      "File bin.60_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.45_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.183_plsdb.m8\n",
      "File bin.183_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.73_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.189_plsdb.m8\n",
      "File bin.189_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.169_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.178_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.69_plsdb.m8\n",
      "File bin.69_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.25_plsdb.m8\n",
      "File bin.25_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.158_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.41_plsdb.m8\n",
      "File bin.41_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.27_plsdb.m8\n",
      "File bin.27_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.132_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.186_plsdb.m8\n",
      "File bin.186_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.139_plsdb.m8\n",
      "File bin.139_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.143_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.172_plsdb.m8\n",
      "File bin.172_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.5_plsdb.m8\n",
      "File bin.5_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.177_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.117_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.16_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.30_plsdb.m8\n",
      "File bin.30_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.56_plsdb.m8\n",
      "File bin.56_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.61_plsdb.m8\n",
      "File bin.61_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.47_plsdb.m8\n",
      "File bin.47_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.21_plsdb.m8\n",
      "File bin.21_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.155_plsdb.m8\n",
      "File bin.155_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.17_plsdb.m8\n",
      "File bin.17_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.15_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.180_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.173_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.174_plsdb.m8\n",
      "File bin.174_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.144_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.142_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.157_plsdb.m8\n",
      "File bin.157_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.167_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.37_plsdb.m8\n",
      "File bin.37_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.32_plsdb.m8\n",
      "File bin.32_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.24_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.12_plsdb.m8\n",
      "File bin.12_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.77_plsdb.m8\n",
      "File bin.77_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.67_plsdb.m8\n",
      "File bin.67_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.22_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.23_plsdb.m8\n",
      "File bin.23_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.50_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.145_plsdb.m8\n",
      "File bin.145_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.88_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.34_plsdb.m8\n",
      "File bin.34_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.109_plsdb.m8\n",
      "File bin.109_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.35_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.58_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.149_plsdb.m8\n",
      "File bin.149_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.195_plsdb.m8\n",
      "File bin.195_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.3_plsdb.m8\n",
      "File bin.3_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.141_plsdb.m8\n",
      "File bin.141_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.106_plsdb.m8\n",
      "File bin.106_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.53_plsdb.m8\n",
      "File bin.53_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.125_plsdb.m8\n",
      "File bin.125_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.98_plsdb.m8\n",
      "File bin.98_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.137_plsdb.m8\n",
      "File bin.137_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.127_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.65_plsdb.m8\n",
      "File bin.65_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.111_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.44_plsdb.m8\n",
      "File bin.44_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.19_plsdb.m8\n",
      "File bin.19_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.162_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.63_plsdb.m8\n",
      "File bin.63_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.82_plsdb.m8\n",
      "File bin.82_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.9_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.18_plsdb.m8\n",
      "File bin.18_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.26_plsdb.m8\n",
      "File bin.26_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.124_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.128_plsdb.m8\n",
      "File bin.128_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.151_plsdb.m8\n",
      "File bin.151_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.7_plsdb.m8\n",
      "File bin.7_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.188_plsdb.m8\n",
      "File bin.188_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.161_plsdb.m8\n",
      "File bin.161_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.165_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.156_plsdb.m8\n",
      "File bin.156_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.118_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.4_plsdb.m8\n",
      "File bin.4_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.10_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.105_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.114_plsdb.m8\n",
      "File bin.114_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.11_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.133_plsdb.m8\n",
      "File bin.133_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.164_plsdb.m8\n",
      "File bin.164_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.36_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.96_plsdb.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.140_plsdb.m8\n",
      "File bin.140_plsdb.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/plasticDB_mmseq_out/bin.170_plsdb.m8\n",
      "File bin.170_plsdb.m8 is empty, skipping.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>bin_id</th>\n",
       "      <th>orf_name</th>\n",
       "      <th>sub_id</th>\n",
       "      <th>species</th>\n",
       "      <th>PLASTIC_GENE</th>\n",
       "      <th>contig_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>bin.90_k141_2477271_length_4624_cov_11.0029_3</td>\n",
       "      <td>bin.90</td>\n",
       "      <td>k141_2477271_length_4624_cov_11.0029_3</td>\n",
       "      <td>00160||PHA_depolymerase||Cupriavidus_necator||...</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>PHA_PHB</td>\n",
       "      <td>bin.90_k141_2477271</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>bin.90_k141_4893777_length_2696_cov_10.4325_2</td>\n",
       "      <td>bin.90</td>\n",
       "      <td>k141_4893777_length_2696_cov_10.4325_2</td>\n",
       "      <td>00161||PHA_depolymerase||Cupriavidus_necator||...</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>PHA_PHB</td>\n",
       "      <td>bin.90_k141_4893777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>bin.90_k141_9840046_length_4167_cov_14.0169_2</td>\n",
       "      <td>bin.90</td>\n",
       "      <td>k141_9840046_length_4167_cov_14.0169_2</td>\n",
       "      <td>00071||3HV_dehydrogenase||Paracoccus_denitrifi...</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>P3HV_PHBV_PHA</td>\n",
       "      <td>bin.90_k141_9840046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>bin.62_k141_8160052_length_3539_cov_9.0000_3</td>\n",
       "      <td>bin.62</td>\n",
       "      <td>k141_8160052_length_3539_cov_9.0000_3</td>\n",
       "      <td>00071||3HV_dehydrogenase||Paracoccus_denitrifi...</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>P3HV_PHBV_PHA</td>\n",
       "      <td>bin.62_k141_8160052</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>bin.110_k141_1989103_length_40976_cov_26.7137_20</td>\n",
       "      <td>bin.110</td>\n",
       "      <td>k141_1989103_length_40976_cov_26.7137_20</td>\n",
       "      <td>00164||PHA_depolymerase||Cupriavidus_necator||...</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>PHA_PHB</td>\n",
       "      <td>bin.110_k141_1989103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>bin.36_k141_7487470_length_7688_cov_9.0000_3</td>\n",
       "      <td>bin.36</td>\n",
       "      <td>k141_7487470_length_7688_cov_9.0000_3</td>\n",
       "      <td>00071||3HV_dehydrogenase||Paracoccus_denitrifi...</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>P3HV_PHBV_PHA</td>\n",
       "      <td>bin.36_k141_7487470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>bin.36_k141_7744773_length_4289_cov_7.8566_2</td>\n",
       "      <td>bin.36</td>\n",
       "      <td>k141_7744773_length_4289_cov_7.8566_2</td>\n",
       "      <td>00161||PHA_depolymerase||Cupriavidus_necator||...</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>PHA_PHB</td>\n",
       "      <td>bin.36_k141_7744773</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>bin.36_k141_9866550_length_2739_cov_13.0000_2</td>\n",
       "      <td>bin.36</td>\n",
       "      <td>k141_9866550_length_2739_cov_13.0000_2</td>\n",
       "      <td>00053||PHB_depolymerase||Diaphorobacter_sp.||P...</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>PHB_PHBV_PHA</td>\n",
       "      <td>bin.36_k141_9866550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>bin.96_k141_10050135_length_1922_cov_8.0000_2</td>\n",
       "      <td>bin.96</td>\n",
       "      <td>k141_10050135_length_1922_cov_8.0000_2</td>\n",
       "      <td>00087||Polyesterase||Pseudomonas_pseudoalcalig...</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>PBAT</td>\n",
       "      <td>bin.96_k141_10050135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>bin.96_k141_7605246_length_3434_cov_12.0000_2</td>\n",
       "      <td>bin.96</td>\n",
       "      <td>k141_7605246_length_3434_cov_12.0000_2</td>\n",
       "      <td>00167||PHA_depolymerase||Cupriavidus_necator||...</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>PHA_PHB</td>\n",
       "      <td>bin.96_k141_7605246</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>133 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   id   bin_id  \\\n",
       "0       bin.90_k141_2477271_length_4624_cov_11.0029_3   bin.90   \n",
       "1       bin.90_k141_4893777_length_2696_cov_10.4325_2   bin.90   \n",
       "2       bin.90_k141_9840046_length_4167_cov_14.0169_2   bin.90   \n",
       "3        bin.62_k141_8160052_length_3539_cov_9.0000_3   bin.62   \n",
       "4    bin.110_k141_1989103_length_40976_cov_26.7137_20  bin.110   \n",
       "..                                                ...      ...   \n",
       "128      bin.36_k141_7487470_length_7688_cov_9.0000_3   bin.36   \n",
       "129      bin.36_k141_7744773_length_4289_cov_7.8566_2   bin.36   \n",
       "130     bin.36_k141_9866550_length_2739_cov_13.0000_2   bin.36   \n",
       "131     bin.96_k141_10050135_length_1922_cov_8.0000_2   bin.96   \n",
       "132     bin.96_k141_7605246_length_3434_cov_12.0000_2   bin.96   \n",
       "\n",
       "                                     orf_name  \\\n",
       "0      k141_2477271_length_4624_cov_11.0029_3   \n",
       "1      k141_4893777_length_2696_cov_10.4325_2   \n",
       "2      k141_9840046_length_4167_cov_14.0169_2   \n",
       "3       k141_8160052_length_3539_cov_9.0000_3   \n",
       "4    k141_1989103_length_40976_cov_26.7137_20   \n",
       "..                                        ...   \n",
       "128     k141_7487470_length_7688_cov_9.0000_3   \n",
       "129     k141_7744773_length_4289_cov_7.8566_2   \n",
       "130    k141_9866550_length_2739_cov_13.0000_2   \n",
       "131    k141_10050135_length_1922_cov_8.0000_2   \n",
       "132    k141_7605246_length_3434_cov_12.0000_2   \n",
       "\n",
       "                                                sub_id  \\\n",
       "0    00160||PHA_depolymerase||Cupriavidus_necator||...   \n",
       "1    00161||PHA_depolymerase||Cupriavidus_necator||...   \n",
       "2    00071||3HV_dehydrogenase||Paracoccus_denitrifi...   \n",
       "3    00071||3HV_dehydrogenase||Paracoccus_denitrifi...   \n",
       "4    00164||PHA_depolymerase||Cupriavidus_necator||...   \n",
       "..                                                 ...   \n",
       "128  00071||3HV_dehydrogenase||Paracoccus_denitrifi...   \n",
       "129  00161||PHA_depolymerase||Cupriavidus_necator||...   \n",
       "130  00053||PHB_depolymerase||Diaphorobacter_sp.||P...   \n",
       "131  00087||Polyesterase||Pseudomonas_pseudoalcalig...   \n",
       "132  00167||PHA_depolymerase||Cupriavidus_necator||...   \n",
       "\n",
       "                                               species   PLASTIC_GENE  \\\n",
       "0    d__Bacteria;p__Proteobacteria;c__Gammaproteoba...        PHA_PHB   \n",
       "1    d__Bacteria;p__Proteobacteria;c__Gammaproteoba...        PHA_PHB   \n",
       "2    d__Bacteria;p__Proteobacteria;c__Gammaproteoba...  P3HV_PHBV_PHA   \n",
       "3    d__Bacteria;p__Proteobacteria;c__Alphaproteoba...  P3HV_PHBV_PHA   \n",
       "4    d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        PHA_PHB   \n",
       "..                                                 ...            ...   \n",
       "128  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...  P3HV_PHBV_PHA   \n",
       "129  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...        PHA_PHB   \n",
       "130  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...   PHB_PHBV_PHA   \n",
       "131  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...           PBAT   \n",
       "132  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...        PHA_PHB   \n",
       "\n",
       "              contig_name  \n",
       "0     bin.90_k141_2477271  \n",
       "1     bin.90_k141_4893777  \n",
       "2     bin.90_k141_9840046  \n",
       "3     bin.62_k141_8160052  \n",
       "4    bin.110_k141_1989103  \n",
       "..                    ...  \n",
       "128   bin.36_k141_7487470  \n",
       "129   bin.36_k141_7744773  \n",
       "130   bin.36_k141_9866550  \n",
       "131  bin.96_k141_10050135  \n",
       "132   bin.96_k141_7605246  \n",
       "\n",
       "[133 rows x 7 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# PlasticDB注释目录\n",
    "plastic_dir = os.path.join('/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation',sample_type+'_bin','plasticDB_mmseq_out')  # 替换为实际路径\n",
    "\n",
    "plastic_metadata = []\n",
    "\n",
    "# 遍历塑料降解基因的注释文件\n",
    "for plastic_file in os.listdir(plastic_dir):\n",
    "    if plastic_file.endswith('.m8'):\n",
    "        bin_num = plastic_file.split('_')[0]\n",
    "        plastic_file_path = os.path.join(plastic_dir, plastic_file)\n",
    "        print(plastic_file_path)\n",
    "        \n",
    "        # 检查文件是否为空\n",
    "        if os.path.getsize(plastic_file_path) == 0:\n",
    "            print(f\"File {plastic_file} is empty, skipping.\")\n",
    "            continue\n",
    "        \n",
    "        # 读取塑料降解基因的注释文件\n",
    "        try:\n",
    "            df = pd.read_csv(plastic_file_path, sep=\"\\t\", header=None)\n",
    "        except pd.errors.EmptyDataError:\n",
    "            print(f\"File {plastic_file} is empty or cannot be parsed, skipping.\")\n",
    "            continue\n",
    "        \n",
    "        # 指定列名\n",
    "        df.columns = ['id', 'sub_id', 'identity', 'alignLen', 'mismat', 'gapOpens', 'qStart', 'qEnd', 'sStart', 'sEnd', 'eval', 'bit']\n",
    "        \n",
    "        # 过滤出符合条件的塑料降解基因注释：identity > 0.6 且 alignLen > 40\n",
    "        filter_df = df[(df['identity'] > 0.5) & (df['alignLen'] > 40)]\n",
    "        \n",
    "        # 去重操作：对于每个 id，选择 identity 最大的行\n",
    "        filter_df = filter_df.loc[filter_df.groupby('id')['identity'].idxmax()]\n",
    "        \n",
    "        # 处理剩下的行并构建结果\n",
    "        for _, row in filter_df.iterrows():\n",
    "            entry_id = f\"{bin_num}_{row['id']}\"\n",
    "            contig_name = bin_num + \"_\" + entry_id.split('_')[1] + \"_\" + entry_id.split('_')[2]\n",
    "            \n",
    "            # 提取 sub_id 中的最后一个字段（塑料降解基因名称）\n",
    "            plastic_gene_name = row['sub_id'].split('||')[-1]\n",
    "            \n",
    "            plastic_metadata.append({\n",
    "                'id': entry_id,\n",
    "                'bin_id': bin_num,\n",
    "                'orf_name': row['id'],\n",
    "                'sub_id': row['sub_id'],\n",
    "                'species': species_dict.get(bin_num, 'Unknown'),\n",
    "                'PLASTIC_GENE': plastic_gene_name,  # 使用提取的塑料降解基因名称\n",
    "                'contig_name': contig_name\n",
    "            })\n",
    "\n",
    "# 将塑料降解基因注释信息转换为 DataFrame\n",
    "plastic_df = pd.DataFrame(plastic_metadata)\n",
    "plastic_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 处理MobileOG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.189_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.117_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.67_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.124_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.99_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.176_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.32_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.57_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.83_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.72_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.155_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.188_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.56_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.118_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.4_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.77_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.148_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.59_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.12_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.167_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.147_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.21_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.60_mobileOG_mmseq2.m8\n",
      "File bin.60_mobileOG_mmseq2.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.98_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.20_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.175_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.7_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.128_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.5_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.123_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.80_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.38_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.81_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.122_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.107_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.22_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.74_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.114_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.182_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.149_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.111_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.33_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.92_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.49_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.174_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.179_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.165_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.130_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.142_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.171_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.184_mobileOG_mmseq2.m8\n",
      "File bin.184_mobileOG_mmseq2.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.160_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.82_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.193_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.97_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.143_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.125_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.15_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.150_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.152_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.69_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.61_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.131_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.46_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.105_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.27_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.39_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.45_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.76_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.186_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.16_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.121_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.138_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.101_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.17_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.90_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.109_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.11_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.42_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.170_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.113_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.162_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.133_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.102_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.78_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.161_mobileOG_mmseq2.m8\n",
      "File bin.161_mobileOG_mmseq2.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.196_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.34_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.86_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.110_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.192_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.6_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.89_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.120_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.30_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.51_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.168_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.177_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.166_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.132_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.62_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.106_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.151_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.158_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.36_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.157_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.8_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.43_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.63_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.40_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.25_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.172_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.37_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.84_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.65_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.28_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.79_mobileOG_mmseq2.m8\n",
      "File bin.79_mobileOG_mmseq2.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.159_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.94_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.75_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.104_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.68_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.96_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.14_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.95_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.163_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.144_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.50_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.91_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.44_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.73_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.29_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.141_mobileOG_mmseq2.m8\n",
      "File bin.141_mobileOG_mmseq2.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.71_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.169_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.26_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.35_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.108_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.103_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.137_mobileOG_mmseq2.m8\n",
      "File bin.137_mobileOG_mmseq2.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.136_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.178_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.173_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.156_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.24_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.13_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.135_mobileOG_mmseq2.m8\n",
      "File bin.135_mobileOG_mmseq2.m8 is empty, skipping.\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.126_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.93_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.55_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.23_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.54_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.139_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.127_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.134_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.41_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.10_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.145_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.52_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.64_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.164_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.85_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.53_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.18_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.100_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.1_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.19_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.48_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.146_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.58_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.183_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.119_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.9_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.195_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.116_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.181_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.31_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.115_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.70_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.153_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.154_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.129_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.197_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.180_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.47_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.2_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.140_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.3_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.87_mobileOG_mmseq2.m8\n",
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/mobileOG_out/bin.88_mobileOG_mmseq2.m8\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>bin_id</th>\n",
       "      <th>orf_name</th>\n",
       "      <th>species</th>\n",
       "      <th>mobileOG_ID</th>\n",
       "      <th>MGE_Name</th>\n",
       "      <th>MGE_class</th>\n",
       "      <th>MGE_Major_Category</th>\n",
       "      <th>MGE_Database</th>\n",
       "      <th>contig_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>bin.189_k141_2766875_length_1634_cov_12.8399_2</td>\n",
       "      <td>bin.189</td>\n",
       "      <td>k141_2766875_length_1634_cov_12.8399_2</td>\n",
       "      <td>d__Bacteria;p__Desulfobacterota_B;c__Binatia;o...</td>\n",
       "      <td>mobileOG_000057348</td>\n",
       "      <td>S4062</td>\n",
       "      <td>integration/excision</td>\n",
       "      <td>NA</td>\n",
       "      <td>COMPASS</td>\n",
       "      <td>bin.189_k141_2766875</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>bin.189_k141_6474206_length_2126_cov_6.0000_2</td>\n",
       "      <td>bin.189</td>\n",
       "      <td>k141_6474206_length_2126_cov_6.0000_2</td>\n",
       "      <td>d__Bacteria;p__Desulfobacterota_B;c__Binatia;o...</td>\n",
       "      <td>mobileOG_000282639</td>\n",
       "      <td>recA</td>\n",
       "      <td>replication/recombination/repair</td>\n",
       "      <td>NA</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>bin.189_k141_6474206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>bin.189_k141_9635482_length_3266_cov_7.0000_1</td>\n",
       "      <td>bin.189</td>\n",
       "      <td>k141_9635482_length_3266_cov_7.0000_1</td>\n",
       "      <td>d__Bacteria;p__Desulfobacterota_B;c__Binatia;o...</td>\n",
       "      <td>mobileOG_000231079</td>\n",
       "      <td>uvrA</td>\n",
       "      <td>replication/recombination/repair</td>\n",
       "      <td>NA</td>\n",
       "      <td>GPD</td>\n",
       "      <td>bin.189_k141_9635482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>bin.117_k141_10181993_length_10614_cov_22.9657_7</td>\n",
       "      <td>bin.117</td>\n",
       "      <td>k141_10181993_length_10614_cov_22.9657_7</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>mobileOG_000040226</td>\n",
       "      <td>trp1400A</td>\n",
       "      <td>integration/excision</td>\n",
       "      <td>NA</td>\n",
       "      <td>COMPASS</td>\n",
       "      <td>bin.117_k141_10181993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>bin.117_k141_2504969_length_15247_cov_12.3442_6</td>\n",
       "      <td>bin.117</td>\n",
       "      <td>k141_2504969_length_15247_cov_12.3442_6</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>mobileOG_000740027</td>\n",
       "      <td>NA:Keyword</td>\n",
       "      <td>transfer</td>\n",
       "      <td>NA</td>\n",
       "      <td>Plasmid</td>\n",
       "      <td>bin.117_k141_2504969</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2146</th>\n",
       "      <td>bin.88_k141_7190286_length_1952_cov_14.0000_2</td>\n",
       "      <td>bin.88</td>\n",
       "      <td>k141_7190286_length_1952_cov_14.0000_2</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>mobileOG_000367080</td>\n",
       "      <td>dut</td>\n",
       "      <td>transfer</td>\n",
       "      <td>regulation</td>\n",
       "      <td>Plasmid</td>\n",
       "      <td>bin.88_k141_7190286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2147</th>\n",
       "      <td>bin.88_k141_9080213_length_1556_cov_11.0000_2</td>\n",
       "      <td>bin.88</td>\n",
       "      <td>k141_9080213_length_1556_cov_11.0000_2</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>mobileOG_000340056</td>\n",
       "      <td>nth</td>\n",
       "      <td>replication/recombination/repair</td>\n",
       "      <td>NA</td>\n",
       "      <td>Plasmid</td>\n",
       "      <td>bin.88_k141_9080213</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2148</th>\n",
       "      <td>bin.88_k141_9097861_length_3149_cov_12.0000_4</td>\n",
       "      <td>bin.88</td>\n",
       "      <td>k141_9097861_length_3149_cov_12.0000_4</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>mobileOG_000302947</td>\n",
       "      <td>stbC</td>\n",
       "      <td>stability/transfer/defense</td>\n",
       "      <td>NA</td>\n",
       "      <td>Plasmid</td>\n",
       "      <td>bin.88_k141_9097861</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2149</th>\n",
       "      <td>bin.88_k141_9097861_length_3149_cov_12.0000_5</td>\n",
       "      <td>bin.88</td>\n",
       "      <td>k141_9097861_length_3149_cov_12.0000_5</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>mobileOG_000051360</td>\n",
       "      <td>vapC</td>\n",
       "      <td>stability/transfer/defense</td>\n",
       "      <td>NA</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>bin.88_k141_9097861</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2150</th>\n",
       "      <td>bin.88_k141_9784439_length_5181_cov_14.7567_4</td>\n",
       "      <td>bin.88</td>\n",
       "      <td>k141_9784439_length_5181_cov_14.7567_4</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>mobileOG_000749623</td>\n",
       "      <td>NA:Keyword</td>\n",
       "      <td>integration/excision</td>\n",
       "      <td>NA</td>\n",
       "      <td>Plasmid</td>\n",
       "      <td>bin.88_k141_9784439</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2151 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    id   bin_id  \\\n",
       "0       bin.189_k141_2766875_length_1634_cov_12.8399_2  bin.189   \n",
       "1        bin.189_k141_6474206_length_2126_cov_6.0000_2  bin.189   \n",
       "2        bin.189_k141_9635482_length_3266_cov_7.0000_1  bin.189   \n",
       "3     bin.117_k141_10181993_length_10614_cov_22.9657_7  bin.117   \n",
       "4      bin.117_k141_2504969_length_15247_cov_12.3442_6  bin.117   \n",
       "...                                                ...      ...   \n",
       "2146     bin.88_k141_7190286_length_1952_cov_14.0000_2   bin.88   \n",
       "2147     bin.88_k141_9080213_length_1556_cov_11.0000_2   bin.88   \n",
       "2148     bin.88_k141_9097861_length_3149_cov_12.0000_4   bin.88   \n",
       "2149     bin.88_k141_9097861_length_3149_cov_12.0000_5   bin.88   \n",
       "2150     bin.88_k141_9784439_length_5181_cov_14.7567_4   bin.88   \n",
       "\n",
       "                                      orf_name  \\\n",
       "0       k141_2766875_length_1634_cov_12.8399_2   \n",
       "1        k141_6474206_length_2126_cov_6.0000_2   \n",
       "2        k141_9635482_length_3266_cov_7.0000_1   \n",
       "3     k141_10181993_length_10614_cov_22.9657_7   \n",
       "4      k141_2504969_length_15247_cov_12.3442_6   \n",
       "...                                        ...   \n",
       "2146    k141_7190286_length_1952_cov_14.0000_2   \n",
       "2147    k141_9080213_length_1556_cov_11.0000_2   \n",
       "2148    k141_9097861_length_3149_cov_12.0000_4   \n",
       "2149    k141_9097861_length_3149_cov_12.0000_5   \n",
       "2150    k141_9784439_length_5181_cov_14.7567_4   \n",
       "\n",
       "                                                species         mobileOG_ID  \\\n",
       "0     d__Bacteria;p__Desulfobacterota_B;c__Binatia;o...  mobileOG_000057348   \n",
       "1     d__Bacteria;p__Desulfobacterota_B;c__Binatia;o...  mobileOG_000282639   \n",
       "2     d__Bacteria;p__Desulfobacterota_B;c__Binatia;o...  mobileOG_000231079   \n",
       "3     d__Bacteria;p__Proteobacteria;c__Alphaproteoba...  mobileOG_000040226   \n",
       "4     d__Bacteria;p__Proteobacteria;c__Alphaproteoba...  mobileOG_000740027   \n",
       "...                                                 ...                 ...   \n",
       "2146  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...  mobileOG_000367080   \n",
       "2147  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...  mobileOG_000340056   \n",
       "2148  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...  mobileOG_000302947   \n",
       "2149  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...  mobileOG_000051360   \n",
       "2150  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...  mobileOG_000749623   \n",
       "\n",
       "        MGE_Name                         MGE_class MGE_Major_Category  \\\n",
       "0          S4062              integration/excision                 NA   \n",
       "1           recA  replication/recombination/repair                 NA   \n",
       "2           uvrA  replication/recombination/repair                 NA   \n",
       "3       trp1400A              integration/excision                 NA   \n",
       "4     NA:Keyword                          transfer                 NA   \n",
       "...          ...                               ...                ...   \n",
       "2146         dut                          transfer         regulation   \n",
       "2147         nth  replication/recombination/repair                 NA   \n",
       "2148        stbC        stability/transfer/defense                 NA   \n",
       "2149        vapC        stability/transfer/defense                 NA   \n",
       "2150  NA:Keyword              integration/excision                 NA   \n",
       "\n",
       "     MGE_Database            contig_name  \n",
       "0         COMPASS   bin.189_k141_2766875  \n",
       "1        Multiple   bin.189_k141_6474206  \n",
       "2             GPD   bin.189_k141_9635482  \n",
       "3         COMPASS  bin.117_k141_10181993  \n",
       "4         Plasmid   bin.117_k141_2504969  \n",
       "...           ...                    ...  \n",
       "2146      Plasmid    bin.88_k141_7190286  \n",
       "2147      Plasmid    bin.88_k141_9080213  \n",
       "2148      Plasmid    bin.88_k141_9097861  \n",
       "2149     Multiple    bin.88_k141_9097861  \n",
       "2150      Plasmid    bin.88_k141_9784439  \n",
       "\n",
       "[2151 rows x 10 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# MobileOG注释目录\n",
    "mge_dir = os.path.join('/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation',sample_type+'_bin','mobileOG_out')  # 替换为实际路径\n",
    "\n",
    "mge_metadata = []\n",
    "\n",
    "# 遍历MGE的注释文件\n",
    "for mge_file in os.listdir(mge_dir):\n",
    "    if mge_file.endswith('.m8'):\n",
    "        bin_num = mge_file.split('_')[0]\n",
    "        mge_file_path = os.path.join(mge_dir, mge_file)\n",
    "        print(mge_file_path)\n",
    "        \n",
    "        # 检查文件是否为空\n",
    "        if os.path.getsize(mge_file_path) == 0:\n",
    "            print(f\"File {mge_file} is empty, skipping.\")\n",
    "            continue\n",
    "        \n",
    "        # 读取MGE注释文件\n",
    "        try:\n",
    "            df_MobileOG = pd.read_csv(mge_file_path, sep=\"\\t\", header=None)\n",
    "        except pd.errors.EmptyDataError:\n",
    "            print(f\"File {mge_file} is empty or cannot be parsed, skipping.\")\n",
    "            continue\n",
    "        \n",
    "        # 指定列名\n",
    "        df_MobileOG.columns = ['id', 'sub_id', 'identity', 'alignLen', 'mismat', 'gapOpens', 'qStart', 'qEnd', 'sStart', 'sEnd', 'eval', 'bit']\n",
    "        \n",
    "        # 处理sub_id\n",
    "        df_MobileOG_tmp = df_MobileOG[\"sub_id\"].str.split(\"|\", expand=True)\n",
    "        \n",
    "        # 如果列数少于7，填充缺失列\n",
    "        if df_MobileOG_tmp.shape[1] < 7:\n",
    "            if df_MobileOG_tmp.shape[1] == 5:\n",
    "                df_MobileOG_tmp[5] = \"-\"\n",
    "            if df_MobileOG_tmp.shape[1] == 6:\n",
    "                df_MobileOG_tmp[6] = \"-\"\n",
    "        \n",
    "        # 选择需要的列并重命名\n",
    "        df_MobileOG_tmp = df_MobileOG_tmp[[0, 1, 3, 4, 5]]\n",
    "        df_MobileOG_tmp.columns = [\"mobileOG_ID\", \"Gene_Name\", \"Taxonomy\", \"Major_Category\", \"MGE_Database\"]\n",
    "        \n",
    "        # 合并处理后的sub_id信息\n",
    "        df_MobileOG_concat = pd.concat((df_MobileOG, df_MobileOG_tmp), axis=1)\n",
    "        \n",
    "        # 过滤出符合条件的MGE注释：identity > 0.6 且 alignLen > 40\n",
    "        filter_df = df_MobileOG_concat[(df_MobileOG_concat['identity'] > 0.6) & (df_MobileOG_concat['alignLen'] > 40)]\n",
    "        \n",
    "        # 去重操作：对于每个 id，选择 identity 最大的行\n",
    "        filter_df = filter_df.loc[filter_df.groupby('id')['identity'].idxmax()]\n",
    "        \n",
    "        # 处理剩下的行并构建结果\n",
    "        for _, row in filter_df.iterrows():\n",
    "            entry_id = f\"{bin_num}_{row['id']}\"\n",
    "            contig_name = bin_num + \"_\" + entry_id.split('_')[1] + \"_\" + entry_id.split('_')[2]\n",
    "            \n",
    "            mge_metadata.append({\n",
    "                'id': entry_id,\n",
    "                'bin_id': bin_num,\n",
    "                'orf_name': row['id'],\n",
    "                'species': species_dict.get(bin_num, 'Unknown'),\n",
    "                'mobileOG_ID': row['mobileOG_ID'],\n",
    "                'MGE_Name': row['Gene_Name'],\n",
    "                'MGE_class': row['Taxonomy'],\n",
    "                'MGE_Major_Category': row['Major_Category'],\n",
    "                'MGE_Database': row['MGE_Database'],\n",
    "                'contig_name': contig_name\n",
    "            })\n",
    "\n",
    "# 将MGE注释信息转换为DataFrame\n",
    "mge_df = pd.DataFrame(mge_metadata)\n",
    "mge_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 合并所有"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/nonbio_bin/merge_result/metadata_arg_vf_plastic_mge.csv\n"
     ]
    }
   ],
   "source": [
    "# 合并ARG与毒力因子注释\n",
    "merged_df_arg_vf = pd.merge(arg_df, vf_df, on=['id', 'contig_name', 'bin_id', 'orf_name','species'], how='outer')\n",
    "\n",
    "# 填充空值为'-'\n",
    "merged_df_arg_vf.fillna('-', inplace=True)\n",
    "\n",
    "# 再合并塑料降解基因注释\n",
    "merged_df_arg_vf_plastic = pd.merge(merged_df_arg_vf, plastic_df, on=['id', 'contig_name', 'bin_id', 'orf_name','species'], how='outer')\n",
    "\n",
    "# 填充空值为'-'\n",
    "merged_df_arg_vf_plastic.fillna('-', inplace=True)\n",
    "\n",
    "# 最后合并MGE注释\n",
    "merged_df_all = pd.merge(merged_df_arg_vf_plastic, mge_df, on=['id', 'contig_name', 'bin_id', 'orf_name','species'], how='outer')\n",
    "\n",
    "# 填充空值为'-'\n",
    "merged_df_all.fillna('-', inplace=True)\n",
    "\n",
    "# 保存最终的metadata表\n",
    "output_dir=os.path.join('/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation',sample_type+'_bin','merge_result')\n",
    "output_file = os.path.join(output_dir,'metadata_arg_vf_plastic_mge.csv')\n",
    "print(output_file)\n",
    "if not os.path.exists(output_dir):\n",
    "    os.makedirs(output_dir)\n",
    "merged_df_all.to_csv(output_file, index=False)\n",
    "\n",
    "# # 打印结果以检查\n",
    "# print(merged_df_all)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 加入orf的起始位置"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Columns in merged_df_all: Index(['id', 'bin_id', 'orf_name', 'species', 'ARG', 'predicted_ARG_class',\n",
      "       'contig_name', 'sub_id_x', 'VF_NAME', 'sub_id_y', 'PLASTIC_GENE',\n",
      "       'mobileOG_ID', 'MGE_Name', 'MGE_class', 'MGE_Major_Category',\n",
      "       'MGE_Database'],\n",
      "      dtype='object')\n",
      "Columns in start_end_df: Index(['orf_name', 'start', 'end', 'strand'], dtype='object')\n",
      "Number of unmatched orf_name: 0\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>bin_id</th>\n",
       "      <th>orf_name</th>\n",
       "      <th>species</th>\n",
       "      <th>ARG</th>\n",
       "      <th>predicted_ARG_class</th>\n",
       "      <th>contig_name</th>\n",
       "      <th>sub_id_x</th>\n",
       "      <th>VF_NAME</th>\n",
       "      <th>sub_id_y</th>\n",
       "      <th>PLASTIC_GENE</th>\n",
       "      <th>mobileOG_ID</th>\n",
       "      <th>MGE_Name</th>\n",
       "      <th>MGE_class</th>\n",
       "      <th>MGE_Major_Category</th>\n",
       "      <th>MGE_Database</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>bin.100_k141_10009953_length_3003_cov_10.0000_2</td>\n",
       "      <td>bin.100</td>\n",
       "      <td>k141_10009953_length_3003_cov_10.0000_2</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.100_k141_10009953</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>mobileOG_000199360</td>\n",
       "      <td>hfq</td>\n",
       "      <td>phage</td>\n",
       "      <td>infection,regulation</td>\n",
       "      <td>GPD</td>\n",
       "      <td>543</td>\n",
       "      <td>782</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>bin.100_k141_5044243_length_4663_cov_10.0000_5</td>\n",
       "      <td>bin.100</td>\n",
       "      <td>k141_5044243_length_4663_cov_10.0000_5</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.100_k141_5044243</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>mobileOG_000334520</td>\n",
       "      <td>thyA</td>\n",
       "      <td>phage</td>\n",
       "      <td>replication,infection,regulation</td>\n",
       "      <td>Plasmid</td>\n",
       "      <td>3888</td>\n",
       "      <td>4661</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>bin.100_k141_5181451_length_1829_cov_5.0000_2</td>\n",
       "      <td>bin.100</td>\n",
       "      <td>k141_5181451_length_1829_cov_5.0000_2</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>PMRF</td>\n",
       "      <td>peptide</td>\n",
       "      <td>bin.100_k141_5181451</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>814</td>\n",
       "      <td>1818</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>bin.100_k141_71747_length_11945_cov_10.0000_10</td>\n",
       "      <td>bin.100</td>\n",
       "      <td>k141_71747_length_11945_cov_10.0000_10</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.100_k141_71747</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>mobileOG_000334176</td>\n",
       "      <td>radA</td>\n",
       "      <td>replication/recombination/repair</td>\n",
       "      <td>competence</td>\n",
       "      <td>Plasmid</td>\n",
       "      <td>8826</td>\n",
       "      <td>10190</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>bin.100_k141_8412174_length_1605_cov_8.0034_2</td>\n",
       "      <td>bin.100</td>\n",
       "      <td>k141_8412174_length_1605_cov_8.0034_2</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>GOLS</td>\n",
       "      <td>multidrug</td>\n",
       "      <td>bin.100_k141_8412174</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>251</td>\n",
       "      <td>694</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2540</th>\n",
       "      <td>bin.9_k141_8293951_length_9442_cov_10.2556_2</td>\n",
       "      <td>bin.9</td>\n",
       "      <td>k141_8293951_length_9442_cov_10.2556_2</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.9_k141_8293951</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>mobileOG_000025523</td>\n",
       "      <td>uvrB</td>\n",
       "      <td>replication/recombination/repair</td>\n",
       "      <td>-</td>\n",
       "      <td>Plasmid</td>\n",
       "      <td>1455</td>\n",
       "      <td>3509</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2541</th>\n",
       "      <td>bin.9_k141_8872801_length_6441_cov_11.0000_5</td>\n",
       "      <td>bin.9</td>\n",
       "      <td>k141_8872801_length_6441_cov_11.0000_5</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>BACA</td>\n",
       "      <td>bacitracin</td>\n",
       "      <td>bin.9_k141_8872801</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>4450</td>\n",
       "      <td>5271</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2542</th>\n",
       "      <td>bin.9_k141_9283942_length_2918_cov_14.6104_1</td>\n",
       "      <td>bin.9</td>\n",
       "      <td>k141_9283942_length_2918_cov_14.6104_1</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.9_k141_9283942</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>mobileOG_000049871</td>\n",
       "      <td>groL</td>\n",
       "      <td>phage</td>\n",
       "      <td>replication,chaperone</td>\n",
       "      <td>COMPASS</td>\n",
       "      <td>3</td>\n",
       "      <td>854</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2543</th>\n",
       "      <td>bin.9_k141_9283942_length_2918_cov_14.6104_2</td>\n",
       "      <td>bin.9</td>\n",
       "      <td>k141_9283942_length_2918_cov_14.6104_2</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>bin.9_k141_9283942</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>mobileOG_000279396</td>\n",
       "      <td>groS</td>\n",
       "      <td>phage</td>\n",
       "      <td>structural</td>\n",
       "      <td>Plasmid</td>\n",
       "      <td>895</td>\n",
       "      <td>1185</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2544</th>\n",
       "      <td>bin.9_k141_9283942_length_2918_cov_14.6104_4</td>\n",
       "      <td>bin.9</td>\n",
       "      <td>k141_9283942_length_2918_cov_14.6104_4</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>CAMP-REGULATORY_PROTEIN</td>\n",
       "      <td>unclassified</td>\n",
       "      <td>bin.9_k141_9283942</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>2037</td>\n",
       "      <td>2702</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2545 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   id   bin_id  \\\n",
       "0     bin.100_k141_10009953_length_3003_cov_10.0000_2  bin.100   \n",
       "1      bin.100_k141_5044243_length_4663_cov_10.0000_5  bin.100   \n",
       "2       bin.100_k141_5181451_length_1829_cov_5.0000_2  bin.100   \n",
       "3      bin.100_k141_71747_length_11945_cov_10.0000_10  bin.100   \n",
       "4       bin.100_k141_8412174_length_1605_cov_8.0034_2  bin.100   \n",
       "...                                               ...      ...   \n",
       "2540     bin.9_k141_8293951_length_9442_cov_10.2556_2    bin.9   \n",
       "2541     bin.9_k141_8872801_length_6441_cov_11.0000_5    bin.9   \n",
       "2542     bin.9_k141_9283942_length_2918_cov_14.6104_1    bin.9   \n",
       "2543     bin.9_k141_9283942_length_2918_cov_14.6104_2    bin.9   \n",
       "2544     bin.9_k141_9283942_length_2918_cov_14.6104_4    bin.9   \n",
       "\n",
       "                                     orf_name  \\\n",
       "0     k141_10009953_length_3003_cov_10.0000_2   \n",
       "1      k141_5044243_length_4663_cov_10.0000_5   \n",
       "2       k141_5181451_length_1829_cov_5.0000_2   \n",
       "3      k141_71747_length_11945_cov_10.0000_10   \n",
       "4       k141_8412174_length_1605_cov_8.0034_2   \n",
       "...                                       ...   \n",
       "2540   k141_8293951_length_9442_cov_10.2556_2   \n",
       "2541   k141_8872801_length_6441_cov_11.0000_5   \n",
       "2542   k141_9283942_length_2918_cov_14.6104_1   \n",
       "2543   k141_9283942_length_2918_cov_14.6104_2   \n",
       "2544   k141_9283942_length_2918_cov_14.6104_4   \n",
       "\n",
       "                                                species  \\\n",
       "0     d__Bacteria;p__Proteobacteria;c__Gammaproteoba...   \n",
       "1     d__Bacteria;p__Proteobacteria;c__Gammaproteoba...   \n",
       "2     d__Bacteria;p__Proteobacteria;c__Gammaproteoba...   \n",
       "3     d__Bacteria;p__Proteobacteria;c__Gammaproteoba...   \n",
       "4     d__Bacteria;p__Proteobacteria;c__Gammaproteoba...   \n",
       "...                                                 ...   \n",
       "2540  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...   \n",
       "2541  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...   \n",
       "2542  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...   \n",
       "2543  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...   \n",
       "2544  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...   \n",
       "\n",
       "                          ARG predicted_ARG_class            contig_name  \\\n",
       "0                           -                   -  bin.100_k141_10009953   \n",
       "1                           -                   -   bin.100_k141_5044243   \n",
       "2                        PMRF             peptide   bin.100_k141_5181451   \n",
       "3                           -                   -     bin.100_k141_71747   \n",
       "4                        GOLS           multidrug   bin.100_k141_8412174   \n",
       "...                       ...                 ...                    ...   \n",
       "2540                        -                   -     bin.9_k141_8293951   \n",
       "2541                     BACA          bacitracin     bin.9_k141_8872801   \n",
       "2542                        -                   -     bin.9_k141_9283942   \n",
       "2543                        -                   -     bin.9_k141_9283942   \n",
       "2544  CAMP-REGULATORY_PROTEIN        unclassified     bin.9_k141_9283942   \n",
       "\n",
       "     sub_id_x VF_NAME sub_id_y PLASTIC_GENE         mobileOG_ID MGE_Name  \\\n",
       "0           -       -        -            -  mobileOG_000199360      hfq   \n",
       "1           -       -        -            -  mobileOG_000334520     thyA   \n",
       "2           -       -        -            -                   -        -   \n",
       "3           -       -        -            -  mobileOG_000334176     radA   \n",
       "4           -       -        -            -                   -        -   \n",
       "...       ...     ...      ...          ...                 ...      ...   \n",
       "2540        -       -        -            -  mobileOG_000025523     uvrB   \n",
       "2541        -       -        -            -                   -        -   \n",
       "2542        -       -        -            -  mobileOG_000049871     groL   \n",
       "2543        -       -        -            -  mobileOG_000279396     groS   \n",
       "2544        -       -        -            -                   -        -   \n",
       "\n",
       "                             MGE_class                MGE_Major_Category  \\\n",
       "0                                phage              infection,regulation   \n",
       "1                                phage  replication,infection,regulation   \n",
       "2                                    -                                 -   \n",
       "3     replication/recombination/repair                        competence   \n",
       "4                                    -                                 -   \n",
       "...                                ...                               ...   \n",
       "2540  replication/recombination/repair                                 -   \n",
       "2541                                 -                                 -   \n",
       "2542                             phage             replication,chaperone   \n",
       "2543                             phage                        structural   \n",
       "2544                                 -                                 -   \n",
       "\n",
       "     MGE_Database  start    end strand  \n",
       "0             GPD    543    782      -  \n",
       "1         Plasmid   3888   4661      -  \n",
       "2               -    814   1818      +  \n",
       "3         Plasmid   8826  10190      +  \n",
       "4               -    251    694      -  \n",
       "...           ...    ...    ...    ...  \n",
       "2540      Plasmid   1455   3509      +  \n",
       "2541            -   4450   5271      +  \n",
       "2542      COMPASS      3    854      -  \n",
       "2543      Plasmid    895   1185      -  \n",
       "2544            -   2037   2702      +  \n",
       "\n",
       "[2545 rows x 19 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import re\n",
    "\n",
    "# 加载现有的DataFrame\n",
    "\n",
    "input_file = output_file\n",
    "merged_df_all = pd.read_csv(input_file)\n",
    "\n",
    "# 查看merged_df_all的列名，检查是否有 orf_name\n",
    "print(\"Columns in merged_df_all:\", merged_df_all.columns)\n",
    "\n",
    "# 提取 orf_name 中的 contig_id 和 编号（编号是最后一个下划线后面的部分）\n",
    "merged_df_all['contig_id'] = merged_df_all['orf_name'].apply(lambda x: '_'.join(x.split('_')[:-1]))\n",
    "merged_df_all['orf_number'] = merged_df_all['orf_name'].apply(lambda x: x.split('_')[-1])\n",
    "\n",
    "# 添加 start、end、length 和 strand 位置信息\n",
    "start_end_metadata = []\n",
    "\n",
    "# GFF文件所在目录\n",
    "gff_dir=os.path.join('/storage/jufengLab/luogaoyang/metagenome_project/DSR/1_clean',sample_type+'_all_reads','PREDICT_GENES')\n",
    "\n",
    "# 遍历每个 bin 的 GFF 文件\n",
    "for bin_id in merged_df_all['bin_id'].unique():\n",
    "    # 提取 bin_id 中的数字部分，假设格式为 \"bin.xx\"\n",
    "    bin_num = bin_id.split('.')[1]  # 提取 . 后面的数字部分\n",
    "\n",
    "    # 构建 GFF 文件路径\n",
    "    gff_file = os.path.join(gff_dir, f'bin.{bin_num}.gff')\n",
    "    \n",
    "    if not os.path.exists(gff_file):\n",
    "        print(f\"GFF file for bin {bin_num} not found, skipping.\")\n",
    "        continue\n",
    "\n",
    "    # 读取 GFF 文件\n",
    "    with open(gff_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if not line.startswith('#'):  # 跳过注释行\n",
    "                columns = line.strip().split('\\t')\n",
    "                if len(columns) > 8 and columns[2] == 'CDS':  # 只处理 CDS 行\n",
    "                    contig_name = columns[0]  # 第一列是 contig_name\n",
    "                    start = int(columns[3])  # 第四列是 start\n",
    "                    end = int(columns[4])  # 第五列是 end\n",
    "                    strand = columns[6]  # 第七列是方向，+ 或者 -\n",
    "                    attributes = columns[8]  # 最后一列是 attributes\n",
    "\n",
    "                    # 简化提取 ORF 的 ID，如 \"ID=a_b\"，直接从 columns[8] 中查找 ID\n",
    "                    orf_id_match = re.search(r'ID=([^;]+)', attributes)  # 查找 ID= 后面的内容\n",
    "                    if orf_id_match:\n",
    "                        orf_id = orf_id_match.group(1)  # 获取 ID=a_b 的值\n",
    "                        orf_number = orf_id.split('_')[-1]  # 提取 ORF 编号（b 部分）\n",
    "                        \n",
    "                        # 组合 contig_name 和 orf_number 以匹配 orf_name\n",
    "                        orf_name_in_gff = f\"{contig_name}_{orf_number}\"\n",
    "                        \n",
    "                        # 记录 ORF 的位置信息\n",
    "                        start_end_metadata.append({\n",
    "                            'orf_name': orf_name_in_gff,  # 组合后的 orf_name 与 merged_df_all 中的相匹配\n",
    "                            'start': start,               # start 位置信息\n",
    "                            'end': end,                   # end 位置信息\n",
    "                            'strand': strand              # 方向\n",
    "                        })\n",
    "\n",
    "# 将位置信息转换为 DataFrame\n",
    "start_end_df = pd.DataFrame(start_end_metadata)\n",
    "\n",
    "# 查看 start_end_df 的列名，检查是否正确提取\n",
    "print(\"Columns in start_end_df:\", start_end_df.columns)\n",
    "\n",
    "# 去除前后空格\n",
    "merged_df_all['orf_name'] = merged_df_all['orf_name'].str.strip()\n",
    "start_end_df['orf_name'] = start_end_df['orf_name'].str.strip()\n",
    "\n",
    "# 检查是否存在不匹配的 orf_name\n",
    "unmatched_orf_names = merged_df_all[~merged_df_all['orf_name'].isin(start_end_df['orf_name'])]\n",
    "\n",
    "print(\"Number of unmatched orf_name:\", len(unmatched_orf_names))\n",
    "\n",
    "# 合并位置信息到原始 DataFrame，按照 'orf_name' 匹配\n",
    "merged_df_with_positions = pd.merge(merged_df_all, start_end_df, on='orf_name', how='left')\n",
    "\n",
    "# 填充空值为 '-'\n",
    "merged_df_with_positions.fillna('-', inplace=True)\n",
    "\n",
    "# 删除临时的 'contig_id' 和 'orf_number' 列\n",
    "merged_df_with_positions.drop(columns=['contig_id', 'orf_number'], inplace=True)\n",
    "\n",
    "merged_df_with_positions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                id   bin_id  \\\n",
      "0  bin.100_k141_10009953_length_3003_cov_10.0000_2  bin.100   \n",
      "1   bin.100_k141_5044243_length_4663_cov_10.0000_5  bin.100   \n",
      "2    bin.100_k141_5181451_length_1829_cov_5.0000_2  bin.100   \n",
      "3   bin.100_k141_71747_length_11945_cov_10.0000_10  bin.100   \n",
      "4    bin.100_k141_8412174_length_1605_cov_8.0034_2  bin.100   \n",
      "\n",
      "                                  orf_name  \\\n",
      "0  k141_10009953_length_3003_cov_10.0000_2   \n",
      "1   k141_5044243_length_4663_cov_10.0000_5   \n",
      "2    k141_5181451_length_1829_cov_5.0000_2   \n",
      "3   k141_71747_length_11945_cov_10.0000_10   \n",
      "4    k141_8412174_length_1605_cov_8.0034_2   \n",
      "\n",
      "                                             species   ARG  \\\n",
      "0  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...     -   \n",
      "1  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...     -   \n",
      "2  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...  PMRF   \n",
      "3  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...     -   \n",
      "4  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...  GOLS   \n",
      "\n",
      "  predicted_ARG_class            contig_name sub_id_x VF_NAME sub_id_y  \\\n",
      "0                   -  bin.100_k141_10009953        -       -        -   \n",
      "1                   -   bin.100_k141_5044243        -       -        -   \n",
      "2             peptide   bin.100_k141_5181451        -       -        -   \n",
      "3                   -     bin.100_k141_71747        -       -        -   \n",
      "4           multidrug   bin.100_k141_8412174        -       -        -   \n",
      "\n",
      "  PLASTIC_GENE         mobileOG_ID MGE_Name                         MGE_class  \\\n",
      "0            -  mobileOG_000199360      hfq                             phage   \n",
      "1            -  mobileOG_000334520     thyA                             phage   \n",
      "2            -                   -        -                                 -   \n",
      "3            -  mobileOG_000334176     radA  replication/recombination/repair   \n",
      "4            -                   -        -                                 -   \n",
      "\n",
      "                 MGE_Major_Category MGE_Database  start    end strand  \n",
      "0              infection,regulation          GPD    543    782      -  \n",
      "1  replication,infection,regulation      Plasmid   3888   4661      -  \n",
      "2                                 -            -    814   1818      +  \n",
      "3                        competence      Plasmid   8826  10190      +  \n",
      "4                                 -            -    251    694      -  \n"
     ]
    }
   ],
   "source": [
    "# 保存最终的 metadata 表，包含 start、end 和 strand 信息\n",
    "output_file_with_positions = os.path.join(output_dir,\"metadata_arg_vf_plastic_mge_with_positions.csv\")\n",
    "merged_df_with_positions.to_csv(output_file_with_positions, index=False)\n",
    "\n",
    "# 打印部分结果以检查\n",
    "print(merged_df_with_positions.head())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 整理arg， sepcies，vf，arg class"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bin_id</th>\n",
       "      <th>peptide</th>\n",
       "      <th>multidrug</th>\n",
       "      <th>fosmidomycin</th>\n",
       "      <th>bacitracin</th>\n",
       "      <th>MLS</th>\n",
       "      <th>aminoglycoside</th>\n",
       "      <th>sulfonamide</th>\n",
       "      <th>tetracycline</th>\n",
       "      <th>fluoroquinolone</th>\n",
       "      <th>beta-lactam</th>\n",
       "      <th>triclosan</th>\n",
       "      <th>rifamycin</th>\n",
       "      <th>fosfomycin</th>\n",
       "      <th>diaminopyrimidine</th>\n",
       "      <th>glycopeptide</th>\n",
       "      <th>arg_num</th>\n",
       "      <th>if_vf</th>\n",
       "      <th>genus</th>\n",
       "      <th>plastic_gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>bin.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>bin.10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Novosphingobium</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>bin.100</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>FEB-7</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>bin.101</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Novosphingobium</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>bin.102</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>SG-bin7</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180</th>\n",
       "      <td>bin.95</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>TMP-24</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>181</th>\n",
       "      <td>bin.96</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Rubrivivax</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182</th>\n",
       "      <td>bin.97</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>WLNW01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>bin.98</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PNKF01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>bin.99</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>SXTU01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>185 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      bin_id  peptide  multidrug  fosmidomycin  bacitracin  MLS  \\\n",
       "0      bin.1        0          0             0           0    0   \n",
       "1     bin.10        0          0             0           0    0   \n",
       "2    bin.100        1          1             1           0    0   \n",
       "3    bin.101        0          0             0           0    0   \n",
       "4    bin.102        0          0             0           0    0   \n",
       "..       ...      ...        ...           ...         ...  ...   \n",
       "180   bin.95        0          0             0           0    0   \n",
       "181   bin.96        0          1             0           1    0   \n",
       "182   bin.97        0          0             0           0    0   \n",
       "183   bin.98        0          0             0           0    0   \n",
       "184   bin.99        0          0             0           0    0   \n",
       "\n",
       "     aminoglycoside  sulfonamide  tetracycline  fluoroquinolone  beta-lactam  \\\n",
       "0                 0            0             0                0            0   \n",
       "1                 0            0             0                0            0   \n",
       "2                 0            0             0                0            0   \n",
       "3                 0            0             0                0            0   \n",
       "4                 0            0             0                0            0   \n",
       "..              ...          ...           ...              ...          ...   \n",
       "180               0            0             0                0            0   \n",
       "181               1            0             0                1            0   \n",
       "182               1            0             1                0            0   \n",
       "183               0            0             0                0            0   \n",
       "184               0            0             0                0            0   \n",
       "\n",
       "     triclosan  rifamycin  fosfomycin  diaminopyrimidine  glycopeptide  \\\n",
       "0            0          0           0                  0             0   \n",
       "1            0          0           0                  0             0   \n",
       "2            0          0           0                  0             0   \n",
       "3            0          0           0                  0             0   \n",
       "4            0          0           0                  0             0   \n",
       "..         ...        ...         ...                ...           ...   \n",
       "180          0          0           0                  0             0   \n",
       "181          0          0           0                  0             0   \n",
       "182          0          0           0                  0             0   \n",
       "183          0          0           0                  0             0   \n",
       "184          0          0           0                  0             0   \n",
       "\n",
       "     arg_num  if_vf            genus  plastic_gene  \n",
       "0        0.0    0.0          Unknown             0  \n",
       "1        0.0    1.0  Novosphingobium             1  \n",
       "2        3.0    0.0            FEB-7             0  \n",
       "3        0.0    1.0  Novosphingobium             1  \n",
       "4        0.0    0.0          SG-bin7             0  \n",
       "..       ...    ...              ...           ...  \n",
       "180      0.0    1.0           TMP-24             1  \n",
       "181      8.0    0.0       Rubrivivax             1  \n",
       "182      3.0    0.0           WLNW01             0  \n",
       "183      0.0    0.0           PNKF01             0  \n",
       "184      0.0    0.0           SXTU01             0  \n",
       "\n",
       "[185 rows x 20 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import re\n",
    "\n",
    "# 读取 CSV 文件\n",
    "input_file = output_file_with_positions\n",
    "merged_df = pd.read_csv(input_file)\n",
    "\n",
    "# 添加一列是否为 plastic_gene，基于 PLASTIC_GENE 列的值，使用 0/1 表示\n",
    "merged_df['plastic_gene'] = merged_df['PLASTIC_GENE'].apply(lambda x: 1 if x != '-' else 0)\n",
    "\n",
    "# 提取 unique 的 ARG class，去掉 'unclassified'\n",
    "arg_classes = merged_df['predicted_ARG_class'].unique()\n",
    "arg_classes = [arg_class for arg_class in arg_classes if arg_class != 'unclassified']\n",
    "\n",
    "# 为每个 ARG class 创建一列，判断每个 bin 是否含有该 class，使用 0/1\n",
    "for arg_class in arg_classes:\n",
    "    merged_df[arg_class] = (merged_df['predicted_ARG_class'] == arg_class).astype(int)\n",
    "\n",
    "# 对每个 bin_id 进行汇总，按 bin_id 来统计每种 ARG class 是否存在\n",
    "arg_class_summary = merged_df.groupby('bin_id')[list(arg_classes)].max().reset_index()\n",
    "\n",
    "# 统计每个 bin_id 的 ARG richness （即 ARG 的数量）\n",
    "arg_counts = merged_df[merged_df['ARG'] != '-'].groupby('bin_id').size().reset_index(name='arg_num')\n",
    "\n",
    "# 判断每个 bin_id 是否有 VF 注释，使用 0/1 表示\n",
    "vf_presence = merged_df[merged_df['VF_NAME'] != '-'].groupby('bin_id').size().reset_index(name='vf_count')\n",
    "vf_presence['if_vf'] = (vf_presence['vf_count'] > 0).astype(int)\n",
    "\n",
    "# 提取到 genus 级别的信息，从 species 列中提取 genus 信息\n",
    "def extract_genus(species_str):\n",
    "    match = re.search(r'g__([^;]+)', species_str)\n",
    "    return match.group(1) if match else 'Unknown'\n",
    "\n",
    "merged_df['genus'] = merged_df['species'].apply(extract_genus)\n",
    "\n",
    "# 提取 genus 信息，按 bin_id 汇总\n",
    "genus_summary = merged_df[['bin_id', 'genus']].drop_duplicates()\n",
    "\n",
    "# 提取 plastic_gene 信息，按 bin_id 汇总\n",
    "plastic_gene_summary = merged_df.groupby('bin_id')['plastic_gene'].max().reset_index()\n",
    "\n",
    "# 合并 ARG class、ARG richness、VF、genus 和 plastic_gene 信息\n",
    "bin_summary = pd.merge(arg_class_summary, arg_counts, on='bin_id', how='left')\n",
    "bin_summary = pd.merge(bin_summary, vf_presence[['bin_id', 'if_vf']], on='bin_id', how='left')\n",
    "bin_summary = pd.merge(bin_summary, genus_summary, on='bin_id', how='left')\n",
    "bin_summary = pd.merge(bin_summary, plastic_gene_summary, on='bin_id', how='left')\n",
    "\n",
    "# 填充缺失值为 0\n",
    "bin_summary.fillna(0, inplace=True)\n",
    "bin_summary.drop(columns=['-'], inplace=True)  # 如果有多余的列 '-'\n",
    "\n",
    "# 保存为 CSV 文件\n",
    "output_file=os.path.join(output_dir,sample_type+\"_bin_summary_with_arg_vf_plastic.csv\")\n",
    "bin_summary.to_csv(output_file, index=False)\n",
    "\n",
    "# 打印结果以检查\n",
    "bin_summary"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 统计难降解塑料"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of bins: 185\n",
      "Number of vf bins (if_vf == 1): 32\n",
      "Percentage of vf bins: 17.30%\n",
      "Number of vf bins with ARGs (arg_num != 0): 9\n",
      "Percentage of vf bins with ARGs: 28.12%\n",
      "Number of vf bins with ARGs and degradation genes: 9\n",
      "Percentage of vf bins with ARGs and degradation genes: 28.12%\n",
      "Number of vf bins with degradation genes: 28\n",
      "Percentage of vf bins with degradation genes: 87.50%\n"
     ]
    }
   ],
   "source": [
    "# 创建 DataFrame\n",
    "df = pd.DataFrame(bin_summary)\n",
    "\n",
    "# 1. 统计总的 bin 数量\n",
    "total_bins = len(df)\n",
    "\n",
    "# 2. 统计带有毒力因子 (if_vf == 1) 的 bin 数量\n",
    "vf_bins = df[df['if_vf'] == 1.0]\n",
    "vf_bin_count = len(vf_bins)\n",
    "vf_bin_ratio = vf_bin_count / total_bins * 100  # 占总数的百分比\n",
    "\n",
    "# 3. 统计在 vf bins 中，携带抗性基因 (arg_num != 0) 的 bin 数量\n",
    "vf_with_arg_bins = vf_bins[vf_bins['arg_num'] != 0]\n",
    "vf_with_arg_count = len(vf_with_arg_bins)\n",
    "vf_with_arg_ratio = vf_with_arg_count / vf_bin_count * 100 if vf_bin_count > 0 else 0  # 占 vf bin 的百分比\n",
    "\n",
    "# 4. 统计在 vf bins 中，携带抗性基因且同时携带降解基因 (plastic_gene == 1) 的 bin 数量\n",
    "vf_with_arg_and_degradation_bins = vf_with_arg_bins[vf_with_arg_bins['plastic_gene'] == 1]\n",
    "vf_with_arg_and_degradation_count = len(vf_with_arg_and_degradation_bins)\n",
    "vf_with_arg_and_degradation_ratio = vf_with_arg_and_degradation_count / vf_bin_count * 100 if vf_bin_count > 0 else 0\n",
    "\n",
    "# 5. 统计在 vf bins 中，携带降解基因 (plastic_gene == 1) 的 bin 数量\n",
    "vf_with_degradation_bins = vf_bins[vf_bins['plastic_gene'] == 1]\n",
    "vf_with_degradation_count = len(vf_with_degradation_bins)\n",
    "vf_with_degradation_ratio = vf_with_degradation_count / vf_bin_count * 100 if vf_bin_count > 0 else 0\n",
    "\n",
    "# 输出结果\n",
    "print(f\"Total number of bins: {total_bins}\")\n",
    "print(f\"Number of vf bins (if_vf == 1): {vf_bin_count}\")\n",
    "print(f\"Percentage of vf bins: {vf_bin_ratio:.2f}%\")\n",
    "print(f\"Number of vf bins with ARGs (arg_num != 0): {vf_with_arg_count}\")\n",
    "print(f\"Percentage of vf bins with ARGs: {vf_with_arg_ratio:.2f}%\")\n",
    "print(f\"Number of vf bins with ARGs and degradation genes: {vf_with_arg_and_degradation_count}\")\n",
    "print(f\"Percentage of vf bins with ARGs and degradation genes: {vf_with_arg_and_degradation_ratio:.2f}%\")\n",
    "print(f\"Number of vf bins with degradation genes: {vf_with_degradation_count}\")\n",
    "print(f\"Percentage of vf bins with degradation genes: {vf_with_degradation_ratio:.2f}%\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 统计可降解塑料"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of bins: 129\n",
      "Number of vf bins (if_vf == 1): 38\n",
      "Percentage of vf bins: 29.46%\n",
      "Number of vf bins with ARGs (arg_num != 0): 25\n",
      "Percentage of vf bins with ARGs: 65.79%\n",
      "Number of vf bins with ARGs and degradation genes: 24\n",
      "Percentage of vf bins with ARGs and degradation genes: 63.16%\n",
      "Number of vf bins with degradation genes: 35\n",
      "Percentage of vf bins with degradation genes: 92.11%\n"
     ]
    }
   ],
   "source": [
    "bin_summary_bio=pd.read_csv(\"/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/bio_bin/merge_result/bin_summary_with_arg_vf_plastic.csv\")\n",
    "\n",
    "# 创建 DataFrame\n",
    "df_bio = pd.DataFrame(bin_summary_bio)\n",
    "\n",
    "# 1. 统计总的 bin 数量\n",
    "total_bins = len(df_bio)\n",
    "\n",
    "# 2. 统计带有毒力因子 (if_vf == 1) 的 bin 数量\n",
    "vf_bins = df_bio[df_bio['if_vf'] == 1.0]\n",
    "vf_bin_count = len(vf_bins)\n",
    "vf_bin_ratio = vf_bin_count / total_bins * 100  # 占总数的百分比\n",
    "\n",
    "# 3. 统计在 vf bins 中，携带抗性基因 (arg_num != 0) 的 bin 数量\n",
    "vf_with_arg_bins = vf_bins[vf_bins['arg_num'] != 0]\n",
    "vf_with_arg_count = len(vf_with_arg_bins)\n",
    "vf_with_arg_ratio = vf_with_arg_count / vf_bin_count * 100 if vf_bin_count > 0 else 0  # 占 vf bin 的百分比\n",
    "\n",
    "# 4. 统计在 vf bins 中，携带抗性基因且同时携带降解基因 (plastic_gene == 1) 的 bin 数量\n",
    "vf_with_arg_and_degradation_bins = vf_with_arg_bins[vf_with_arg_bins['plastic_gene'] == 1]\n",
    "vf_with_arg_and_degradation_count = len(vf_with_arg_and_degradation_bins)\n",
    "vf_with_arg_and_degradation_ratio = vf_with_arg_and_degradation_count / vf_bin_count * 100 if vf_bin_count > 0 else 0\n",
    "\n",
    "# 5. 统计在 vf bins 中，携带降解基因 (plastic_gene == 1) 的 bin 数量\n",
    "vf_with_degradation_bins = vf_bins[vf_bins['plastic_gene'] == 1]\n",
    "vf_with_degradation_count = len(vf_with_degradation_bins)\n",
    "vf_with_degradation_ratio = vf_with_degradation_count / vf_bin_count * 100 if vf_bin_count > 0 else 0\n",
    "\n",
    "# 输出结果\n",
    "print(f\"Total number of bins: {total_bins}\")\n",
    "print(f\"Number of vf bins (if_vf == 1): {vf_bin_count}\")\n",
    "print(f\"Percentage of vf bins: {vf_bin_ratio:.2f}%\")\n",
    "print(f\"Number of vf bins with ARGs (arg_num != 0): {vf_with_arg_count}\")\n",
    "print(f\"Percentage of vf bins with ARGs: {vf_with_arg_ratio:.2f}%\")\n",
    "print(f\"Number of vf bins with ARGs and degradation genes: {vf_with_arg_and_degradation_count}\")\n",
    "print(f\"Percentage of vf bins with ARGs and degradation genes: {vf_with_arg_and_degradation_ratio:.2f}%\")\n",
    "print(f\"Number of vf bins with degradation genes: {vf_with_degradation_count}\")\n",
    "print(f\"Percentage of vf bins with degradation genes: {vf_with_degradation_ratio:.2f}%\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 统计水"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of bins: 169\n",
      "Number of vf bins (if_vf == 1): 27\n",
      "Percentage of vf bins: 15.98%\n",
      "Number of vf bins with ARGs (arg_num != 0): 20\n",
      "Percentage of vf bins with ARGs: 74.07%\n",
      "Number of vf bins with ARGs and degradation genes: 16\n",
      "Percentage of vf bins with ARGs and degradation genes: 59.26%\n",
      "Number of vf bins with degradation genes: 22\n",
      "Percentage of vf bins with degradation genes: 81.48%\n"
     ]
    }
   ],
   "source": [
    "bin_summary_water=pd.read_csv(\"/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/water_bin/merge_result/water_bin_summary_with_arg_vf_plastic.csv\")\n",
    "\n",
    "# 创建 DataFrame\n",
    "df_bio = pd.DataFrame(bin_summary_water)\n",
    "\n",
    "# 1. 统计总的 bin 数量\n",
    "total_bins = len(df_bio)\n",
    "\n",
    "# 2. 统计带有毒力因子 (if_vf == 1) 的 bin 数量\n",
    "vf_bins = df_bio[df_bio['if_vf'] == 1.0]\n",
    "vf_bin_count = len(vf_bins)\n",
    "vf_bin_ratio = vf_bin_count / total_bins * 100  # 占总数的百分比\n",
    "\n",
    "# 3. 统计在 vf bins 中，携带抗性基因 (arg_num != 0) 的 bin 数量\n",
    "vf_with_arg_bins = vf_bins[vf_bins['arg_num'] != 0]\n",
    "vf_with_arg_count = len(vf_with_arg_bins)\n",
    "vf_with_arg_ratio = vf_with_arg_count / vf_bin_count * 100 if vf_bin_count > 0 else 0  # 占 vf bin 的百分比\n",
    "\n",
    "# 4. 统计在 vf bins 中，携带抗性基因且同时携带降解基因 (plastic_gene == 1) 的 bin 数量\n",
    "vf_with_arg_and_degradation_bins = vf_with_arg_bins[vf_with_arg_bins['plastic_gene'] == 1]\n",
    "vf_with_arg_and_degradation_count = len(vf_with_arg_and_degradation_bins)\n",
    "vf_with_arg_and_degradation_ratio = vf_with_arg_and_degradation_count / vf_bin_count * 100 if vf_bin_count > 0 else 0\n",
    "\n",
    "# 5. 统计在 vf bins 中，携带降解基因 (plastic_gene == 1) 的 bin 数量\n",
    "vf_with_degradation_bins = vf_bins[vf_bins['plastic_gene'] == 1]\n",
    "vf_with_degradation_count = len(vf_with_degradation_bins)\n",
    "vf_with_degradation_ratio = vf_with_degradation_count / vf_bin_count * 100 if vf_bin_count > 0 else 0\n",
    "\n",
    "# 输出结果\n",
    "print(f\"Total number of bins: {total_bins}\")\n",
    "print(f\"Number of vf bins (if_vf == 1): {vf_bin_count}\")\n",
    "print(f\"Percentage of vf bins: {vf_bin_ratio:.2f}%\")\n",
    "print(f\"Number of vf bins with ARGs (arg_num != 0): {vf_with_arg_count}\")\n",
    "print(f\"Percentage of vf bins with ARGs: {vf_with_arg_ratio:.2f}%\")\n",
    "print(f\"Number of vf bins with ARGs and degradation genes: {vf_with_arg_and_degradation_count}\")\n",
    "print(f\"Percentage of vf bins with ARGs and degradation genes: {vf_with_arg_and_degradation_ratio:.2f}%\")\n",
    "print(f\"Number of vf bins with degradation genes: {vf_with_degradation_count}\")\n",
    "print(f\"Percentage of vf bins with degradation genes: {vf_with_degradation_ratio:.2f}%\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 统计木头"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of bins: 19\n",
      "Number of vf bins (if_vf == 1): 1\n",
      "Percentage of vf bins: 5.26%\n",
      "Number of vf bins with ARGs (arg_num != 0): 1\n",
      "Percentage of vf bins with ARGs: 100.00%\n",
      "Number of vf bins with ARGs and degradation genes: 1\n",
      "Percentage of vf bins with ARGs and degradation genes: 100.00%\n",
      "Number of vf bins with degradation genes: 1\n",
      "Percentage of vf bins with degradation genes: 100.00%\n"
     ]
    }
   ],
   "source": [
    "bin_summary_wood=pd.read_csv(\"/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/wood_bin/merge_result/wood_bin_summary_with_arg_vf_plastic.csv\")\n",
    "\n",
    "# 创建 DataFrame\n",
    "df_bio = pd.DataFrame(bin_summary_wood)\n",
    "\n",
    "# 1. 统计总的 bin 数量\n",
    "total_bins = len(df_bio)\n",
    "\n",
    "# 2. 统计带有毒力因子 (if_vf == 1) 的 bin 数量\n",
    "vf_bins = df_bio[df_bio['if_vf'] == 1.0]\n",
    "vf_bin_count = len(vf_bins)\n",
    "vf_bin_ratio = vf_bin_count / total_bins * 100  # 占总数的百分比\n",
    "\n",
    "# 3. 统计在 vf bins 中，携带抗性基因 (arg_num != 0) 的 bin 数量\n",
    "vf_with_arg_bins = vf_bins[vf_bins['arg_num'] != 0]\n",
    "vf_with_arg_count = len(vf_with_arg_bins)\n",
    "vf_with_arg_ratio = vf_with_arg_count / vf_bin_count * 100 if vf_bin_count > 0 else 0  # 占 vf bin 的百分比\n",
    "\n",
    "# 4. 统计在 vf bins 中，携带抗性基因且同时携带降解基因 (plastic_gene == 1) 的 bin 数量\n",
    "vf_with_arg_and_degradation_bins = vf_with_arg_bins[vf_with_arg_bins['plastic_gene'] == 1]\n",
    "vf_with_arg_and_degradation_count = len(vf_with_arg_and_degradation_bins)\n",
    "vf_with_arg_and_degradation_ratio = vf_with_arg_and_degradation_count / vf_bin_count * 100 if vf_bin_count > 0 else 0\n",
    "\n",
    "# 5. 统计在 vf bins 中，携带降解基因 (plastic_gene == 1) 的 bin 数量\n",
    "vf_with_degradation_bins = vf_bins[vf_bins['plastic_gene'] == 1]\n",
    "vf_with_degradation_count = len(vf_with_degradation_bins)\n",
    "vf_with_degradation_ratio = vf_with_degradation_count / vf_bin_count * 100 if vf_bin_count > 0 else 0\n",
    "\n",
    "# 输出结果\n",
    "print(f\"Total number of bins: {total_bins}\")\n",
    "print(f\"Number of vf bins (if_vf == 1): {vf_bin_count}\")\n",
    "print(f\"Percentage of vf bins: {vf_bin_ratio:.2f}%\")\n",
    "print(f\"Number of vf bins with ARGs (arg_num != 0): {vf_with_arg_count}\")\n",
    "print(f\"Percentage of vf bins with ARGs: {vf_with_arg_ratio:.2f}%\")\n",
    "print(f\"Number of vf bins with ARGs and degradation genes: {vf_with_arg_and_degradation_count}\")\n",
    "print(f\"Percentage of vf bins with ARGs and degradation genes: {vf_with_arg_and_degradation_ratio:.2f}%\")\n",
    "print(f\"Number of vf bins with degradation genes: {vf_with_degradation_count}\")\n",
    "print(f\"Percentage of vf bins with degradation genes: {vf_with_degradation_ratio:.2f}%\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "biopython",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
