{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processed S0PHA.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/filtered_S0PHA.contigs_5M_contigs_fna2faa_all_sos.m8\n",
      "Processed S1PP.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/filtered_S1PP.contigs_5M_contigs_fna2faa_all_sos.m8\n",
      "Processed S0PCL.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/filtered_S0PCL.contigs_5M_contigs_fna2faa_all_sos.m8\n",
      "Processed S2PCL.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/filtered_S2PCL.contigs_5M_contigs_fna2faa_all_sos.m8\n",
      "Processed S3WOOD.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/filtered_S3WOOD.contigs_5M_contigs_fna2faa_all_sos.m8\n",
      "Processed S2WOOD.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/filtered_S2WOOD.contigs_5M_contigs_fna2faa_all_sos.m8\n",
      "Processed S3PCL.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/filtered_S3PCL.contigs_5M_contigs_fna2faa_all_sos.m8\n",
      "Processed S3PHA.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/filtered_S3PHA.contigs_5M_contigs_fna2faa_all_sos.m8\n",
      "Processed S1WOOD.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/filtered_S1WOOD.contigs_5M_contigs_fna2faa_all_sos.m8\n",
      "Processed S0PP.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/filtered_S0PP.contigs_5M_contigs_fna2faa_all_sos.m8\n",
      "Processed S3WE.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/filtered_S3WE.contigs_5M_contigs_fna2faa_all_sos.m8\n",
      "Processed S2PP.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/filtered_S2PP.contigs_5M_contigs_fna2faa_all_sos.m8\n",
      "Processed S0WE.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/filtered_S0WE.contigs_5M_contigs_fna2faa_all_sos.m8\n",
      "Processed S1PCL.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/filtered_S1PCL.contigs_5M_contigs_fna2faa_all_sos.m8\n",
      "Processed S2WE.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/filtered_S2WE.contigs_5M_contigs_fna2faa_all_sos.m8\n",
      "Processed S1WE.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/filtered_S1WE.contigs_5M_contigs_fna2faa_all_sos.m8\n",
      "Processed S2PS.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/filtered_S2PS.contigs_5M_contigs_fna2faa_all_sos.m8\n",
      "Processed S3PP.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/filtered_S3PP.contigs_5M_contigs_fna2faa_all_sos.m8\n",
      "Processed S1PHA.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/filtered_S1PHA.contigs_5M_contigs_fna2faa_all_sos.m8\n",
      "Processed S2PHA.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/filtered_S2PHA.contigs_5M_contigs_fna2faa_all_sos.m8\n",
      "Processed S0WOOD.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/filtered_S0WOOD.contigs_5M_contigs_fna2faa_all_sos.m8\n",
      "Processed S3PS.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/filtered_S3PS.contigs_5M_contigs_fna2faa_all_sos.m8\n",
      "Processed S1PS.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/filtered_S1PS.contigs_5M_contigs_fna2faa_all_sos.m8\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "# 定义路径\n",
    "m8_directory = \"/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out\"  # 修改为你的8m文件路径\n",
    "metadata_file = \"/storage/jufengLab/luogaoyang/db/SOS_ROS_gene_full/combined_metadata.tsv\"  # metadata文件路径\n",
    "group_file = \"/storage/jufengLab/luogaoyang/db/SOS_ROS_gene_full/group.txt\"  # group文件路径\n",
    "\n",
    "# 读取metadata文件\n",
    "metadata = pd.read_csv(metadata_file, sep='\\t')\n",
    "\n",
    "# 检查metadata中是否存在Gene列和Entry列\n",
    "if 'Entry' not in metadata.columns or 'Gene' not in metadata.columns:\n",
    "    raise ValueError(\"Metadata file is missing required 'Entry' or 'Gene' columns.\")\n",
    "\n",
    "# 从metadata中提取需要的列：Entry 和 Gene\n",
    "metadata = metadata[['Entry', 'Gene']]\n",
    "\n",
    "# 读取group文件，提取Gene和其对应的group\n",
    "group_df = pd.read_csv(group_file, sep=\"\\t\", header=0)\n",
    "\n",
    "# 检查group_df中是否存在Gene列\n",
    "if 'Gene' not in group_df.columns:\n",
    "    raise ValueError(\"Group file is missing required 'Gene' column.\")\n",
    "\n",
    "# 遍历目录中的每个8m文件\n",
    "for filename in os.listdir(m8_directory):\n",
    "    if filename.endswith(\".m8\"):\n",
    "        # 获取文件的完整路径\n",
    "        m8_file = os.path.join(m8_directory, filename)\n",
    "        \n",
    "        # 从文件名中提取样本名（假设文件名格式为 sample1.8m）\n",
    "        sample_name = os.path.splitext(filename)[0]  # 去掉扩展名以提取样本名\n",
    "        sample_name= sample_name.split('.')[0]\n",
    "        # 读取8m文件\n",
    "        df = pd.read_csv(m8_file, sep=\"\\t\", header=None)\n",
    "        df.columns = ['id', 'sub_id', 'identity', 'alignLen', 'mismat', 'gapOpens', 'qStart', 'qEnd', 'sStart', 'sEnd', 'eval', 'bit']\n",
    "        \n",
    "        # 过滤条件：identity > 0.7 且 alignLen > 40\n",
    "        filter_df = df[(df['identity'] > 0.6) & (df['alignLen'] > 40)]\n",
    "\n",
    "        # 对每个id，选择identity最高的行\n",
    "        best_hits = filter_df.loc[filter_df.groupby('id')['identity'].idxmax()]\n",
    "\n",
    "        # 检查是否存在 'sub_id' 列，并确保与 metadata 的 'Entry' 列匹配\n",
    "        if best_hits['sub_id'].isnull().any():\n",
    "            raise ValueError(f\"Missing 'sub_id' values in {filename}\")\n",
    "\n",
    "        # 根据sub_id（即Entry）将结果和metadata进行合并，添加Gene信息\n",
    "        merged_df = pd.merge(best_hits, metadata, how='left', left_on='sub_id', right_on='Entry')\n",
    "\n",
    "        # 检查是否成功合并Gene信息\n",
    "        if 'Gene' not in merged_df.columns or merged_df['Gene'].isnull().any():\n",
    "            print(f\"Warning: Some sub_ids in {filename} do not have matching entries in metadata!\")\n",
    "            print(merged_df[['sub_id', 'Entry', 'Gene']].head())  # 打印部分结果调试\n",
    "\n",
    "        # 检查是否存在 'Gene' 列，确保后续合并不会出错\n",
    "        if 'Gene' in merged_df.columns:\n",
    "            # 根据Gene将结果和group_df进行合并，添加group信息\n",
    "            merged_df = pd.merge(merged_df, group_df, how='left', on='Gene')\n",
    "        else:\n",
    "            raise ValueError(f\"'Gene' column is missing after merging with metadata for file {filename}\")\n",
    "\n",
    "        # 添加样本名列\n",
    "        merged_df['Sample'] = sample_name\n",
    "\n",
    "        # 检查是否成功合并group信息\n",
    "        if 'group' not in merged_df.columns or merged_df['group'].isnull().any():\n",
    "            print(f\"Warning: Some Genes in {filename} do not have matching group information!\")\n",
    "\n",
    "        # 输出到新的文件，添加Sample和group列\n",
    "        output_file = os.path.join(m8_directory, \"filter_m8\",f\"filtered_{filename}\")\n",
    "        merged_df.to_csv(output_file, sep='\\t', index=False)\n",
    "\n",
    "        print(f\"Processed {filename} and saved to {output_file}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "加入rpkg信息"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processed filtered_S3PCL.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/with_rpkg_contig/filtered_filtered_S3PCL.contigs_5M_contigs_fna2faa_all_sos_rpkg_60.txt\n",
      "Processed filtered_S2PHA.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/with_rpkg_contig/filtered_filtered_S2PHA.contigs_5M_contigs_fna2faa_all_sos_rpkg_60.txt\n",
      "Processed filtered_S3PHA.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/with_rpkg_contig/filtered_filtered_S3PHA.contigs_5M_contigs_fna2faa_all_sos_rpkg_60.txt\n",
      "Processed filtered_S2PP.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/with_rpkg_contig/filtered_filtered_S2PP.contigs_5M_contigs_fna2faa_all_sos_rpkg_60.txt\n",
      "Processed filtered_S1PCL.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/with_rpkg_contig/filtered_filtered_S1PCL.contigs_5M_contigs_fna2faa_all_sos_rpkg_60.txt\n",
      "Processed filtered_S1WE.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/with_rpkg_contig/filtered_filtered_S1WE.contigs_5M_contigs_fna2faa_all_sos_rpkg_60.txt\n",
      "Processed filtered_S1PHA.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/with_rpkg_contig/filtered_filtered_S1PHA.contigs_5M_contigs_fna2faa_all_sos_rpkg_60.txt\n",
      "Processed filtered_S3WE.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/with_rpkg_contig/filtered_filtered_S3WE.contigs_5M_contigs_fna2faa_all_sos_rpkg_60.txt\n",
      "Processed filtered_S2PS.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/with_rpkg_contig/filtered_filtered_S2PS.contigs_5M_contigs_fna2faa_all_sos_rpkg_60.txt\n",
      "Processed filtered_S2PCL.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/with_rpkg_contig/filtered_filtered_S2PCL.contigs_5M_contigs_fna2faa_all_sos_rpkg_60.txt\n",
      "Processed filtered_S1PP.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/with_rpkg_contig/filtered_filtered_S1PP.contigs_5M_contigs_fna2faa_all_sos_rpkg_60.txt\n",
      "Processed filtered_S3PP.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/with_rpkg_contig/filtered_filtered_S3PP.contigs_5M_contigs_fna2faa_all_sos_rpkg_60.txt\n",
      "Processed filtered_S0WOOD.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/with_rpkg_contig/filtered_filtered_S0WOOD.contigs_5M_contigs_fna2faa_all_sos_rpkg_60.txt\n",
      "Processed filtered_S0WE.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/with_rpkg_contig/filtered_filtered_S0WE.contigs_5M_contigs_fna2faa_all_sos_rpkg_60.txt\n",
      "Processed filtered_S3PS.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/with_rpkg_contig/filtered_filtered_S3PS.contigs_5M_contigs_fna2faa_all_sos_rpkg_60.txt\n",
      "Processed filtered_S2WE.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/with_rpkg_contig/filtered_filtered_S2WE.contigs_5M_contigs_fna2faa_all_sos_rpkg_60.txt\n",
      "Processed filtered_S3WOOD.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/with_rpkg_contig/filtered_filtered_S3WOOD.contigs_5M_contigs_fna2faa_all_sos_rpkg_60.txt\n",
      "Processed filtered_S0PCL.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/with_rpkg_contig/filtered_filtered_S0PCL.contigs_5M_contigs_fna2faa_all_sos_rpkg_60.txt\n",
      "Processed filtered_S1WOOD.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/with_rpkg_contig/filtered_filtered_S1WOOD.contigs_5M_contigs_fna2faa_all_sos_rpkg_60.txt\n",
      "Processed filtered_S0PP.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/with_rpkg_contig/filtered_filtered_S0PP.contigs_5M_contigs_fna2faa_all_sos_rpkg_60.txt\n",
      "Processed filtered_S0PHA.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/with_rpkg_contig/filtered_filtered_S0PHA.contigs_5M_contigs_fna2faa_all_sos_rpkg_60.txt\n",
      "Processed filtered_S1PS.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/with_rpkg_contig/filtered_filtered_S1PS.contigs_5M_contigs_fna2faa_all_sos_rpkg_60.txt\n",
      "Processed filtered_S2WOOD.contigs_5M_contigs_fna2faa_all_sos.m8 and saved to /storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8/with_rpkg_contig/filtered_filtered_S2WOOD.contigs_5M_contigs_fna2faa_all_sos_rpkg_60.txt\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "rpkm_type='contig' #contig or fna\n",
    "filter_id_cutoff=\"60\"\n",
    "\n",
    "# 定义文件路径\n",
    "m8_directory = \"/storage/jufengLab/luogaoyang/metagenome_project/DSR/4_annotation/ROS_gene_mmseq_contig_level_out/sustech_gene_out/filter_m8\"\n",
    "rpkg_directory = \"/storage/jufengLab/luogaoyang/metagenome_project/DSR/PRKG/\"+rpkm_type\n",
    "out_dir=os.path.join(m8_directory,\"with_rpkg_\"+rpkm_type)\n",
    "if not os.path.exists(out_dir):\n",
    "    print(f\"{out_dir} not found!\")\n",
    "    print(f\"makedir {out_dir}...\")\n",
    "    os.makedirs(out_dir)\n",
    "    \n",
    "# 遍历m8文件\n",
    "for filename in os.listdir(m8_directory):\n",
    "    if filename.endswith(\".m8\"):\n",
    "        # 获取文件的完整路径\n",
    "        m8_file = os.path.join(m8_directory, filename)\n",
    "\n",
    "        # 提取sample_name\n",
    "        sample_name = filename.split('_')[1]  # 假设第二部分为sample_name，比如S0PCL\n",
    "        sample_name = sample_name.split('.')[0]\n",
    "\n",
    "        # 对应的RPKG文件路径\n",
    "        rpkg_file = os.path.join(rpkg_directory, f\"{sample_name}_RPKG.csv\")\n",
    "\n",
    "        # 读取RPKG文件\n",
    "        if not os.path.exists(rpkg_file):\n",
    "            print(f\"RPKG file not found for sample {sample_name}. Skipping...\")\n",
    "            continue\n",
    "\n",
    "        rpkg_df = pd.read_csv(rpkg_file)\n",
    "\n",
    "        # 确保RPKG文件包含必要的列\n",
    "        if '#Name' not in rpkg_df.columns or 'RPKG' not in rpkg_df.columns:\n",
    "            print(f\"RPKG file {rpkg_file} is missing required columns. Skipping...\")\n",
    "            continue\n",
    "\n",
    "        # 创建RPKG字典：#Name -> RPKG\n",
    "        rpkg_dict = pd.Series(rpkg_df.RPKG.values, index=rpkg_df['#Name']).to_dict()\n",
    "\n",
    "        # 读取m8文件\n",
    "        m8_df = pd.read_csv(m8_file, sep=\"\\t\", header=0)\n",
    "        m8_df.columns = ['id', 'sub_id', 'identity', 'alignLen', 'mismat', 'gapOpens', 'qStart', 'qEnd',\n",
    "                         'sStart', 'sEnd', 'eval', 'bit', 'Entry', 'Gene', 'group', 'Sample']\n",
    "\n",
    "        # 去除m8文件中id的最后一个下划线后缀\n",
    "        if rpkm_type=='contig':\n",
    "            m8_df['id_modified'] = m8_df['id'].apply(lambda x: '_'.join(x.split('_')[:-2]))\n",
    "        else: #rpkm_type=='fna':\n",
    "            m8_df['id_modified'] = m8_df['id'].apply(lambda x: '_'.join(x.split('_')[:-1]))\n",
    "            \n",
    "\n",
    "        # 查找RPKG值并加入到m8文件中\n",
    "        m8_df['RPKG'] = m8_df['id_modified'].map(rpkg_dict)\n",
    "\n",
    "        # 检查是否有未匹配到的RPKG值\n",
    "        unmatched_ids = m8_df[m8_df['RPKG'].isnull()]\n",
    "        if not unmatched_ids.empty:\n",
    "            print(f\"Warning: {len(unmatched_ids)} ids in {filename} did not match any RPKG values.\")\n",
    "\n",
    "        # 输出新的文件，包含原有列和 RPKG 列\n",
    "        output_filename = f\"filtered_{filename.replace('.m8', '')}_rpkg_{filter_id_cutoff}.txt\"  # 生成新文件名并去掉 .m8 后缀\n",
    "        output_file = os.path.join(m8_directory, \"with_rpkg_\"+rpkm_type,output_filename)\n",
    "        \n",
    "        # 保存文件，避免重复列名\n",
    "        m8_df.to_csv(output_file, sep='\\t', index=False)\n",
    "\n",
    "        print(f\"Processed {filename} and saved to {output_file}\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "biopython",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "undefined.undefined.undefined"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
